---
title: "Association with biomarkers and renal function"
---

```{r setup}
source('0.0.- Setup.R')
```

## Standard Covariates

```{r read_covars}
ukb_gwas_covars <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_gwas_covars.csv'),
  col_types = cols())
```

## Variables for Biomarker Analysis

### Blood Biomarkers

```{r read_blood_biomarker_ids}
ukb_all_biochem_ids <- read_csv(
  file.path(root_fd, 'data', 'releases', 'ukb_ids_biochem_all.csv'),
  col_types = cols())
```

```{r read_blood_biomarkers}
ukb_all_biochem <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_biochem_all.csv'),
  col_types = cols()) |> 
  group_by(eid) |>
  pivot_longer(-eid, names_to = 'id', values_to = 'val') |>
  group_by(id) |>
  separate(id, into = 'id', sep = '-', extra = 'drop') |>
  mutate(id = as.numeric(id)) |>
  inner_join(ukb_all_biochem_ids |> select(id, title), by = 'id') |>
  arrange(eid, title) |>
  select(-id) |>
  pivot_wider(names_from = title, values_from = val)
```

### Blood extraction date

```{r read_blood_extraction_dates}
ukb_blood_extr <- read_csv(
  file.path(root_fd, 'data', 'releases', 'pheno_9.csv'),
         col_types = cols()) |>
  select(eid, Extraction = `3166-0.0`) |>
  filter(!is.na(Extraction)) |>
  mutate(year = lubridate::year(Extraction)) |>
  filter(as.numeric(as.character(year)) > 1900) |>
  select(-year)
```

```{r construct_blood_extraction_covars}
ukb_blood_extr_cov <- ukb_blood_extr |> 
  mutate(time_hour = as.numeric(lubridate::hour(Extraction)),
         time_min = as.numeric(lubridate::minute(Extraction)),
         time_sec = as.numeric(lubridate::second(Extraction)),
         `Extr. Time` = time_hour + time_min/60 + time_sec/3600) |>
  select(-starts_with('time_')) |>
  mutate(`Extr. Month` = case_when(
    lubridate::year(Extraction) == 2006 ~ '2006',
    Extraction >= lubridate::as_date('2010-08-01') ~ '2010_end',
    TRUE ~ paste(lubridate::year(Extraction),
                 lubridate::month(Extraction), sep = '_'))) |>
  select(-Extraction)
```

### Assay date

```{r read_biomarker_processing_field_ids}
ukb_biochem_proc_ids <- read_csv(
  file.path(root_fd, 'data', 'releases', 'ukb_ids_biochem_proc.csv'),
  col_types = cols())
```

```{r read_biomarker_processing}
ukb_biochem_proc <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_biochem_proc.csv'),
  col_types = cols()) |>
  rename_with(~ gsub("-0.0", "", .x, fixed = TRUE)) |>
  rename_with(~ ukb_biochem_proc_ids |>
                filter(id == as.numeric(.x)) |>
                pull(title), -eid)
```

```{r construct_biomarker_processing_covars}
ukb_biochem_proc_cov <- ukb_biochem_proc |>
  select(eid, ends_with('assay date'), contains('dilution')) |>
  mutate(across(where(lubridate::is.Date),
                ~ as.numeric(.x - min(.x, na.rm = TRUE))))
```

### Fasting time

```{r read_fasting_time}
ukb_fasting <- read_csv(
  file.path(root_fd, 'data', 'releases', 'pheno_11.csv'),
  col_types = cols()) |>
  select(eid, `Fasting Time` = `74-0.0`)
```

### Statins

```{r read_statin_med_codes}
ukb_statin_codes <- read_tsv(
  file.path(root_fd, 'data', 'pheno', 'var_4_enc.tsv'),
  col_types = cols()) |>
  filter(meaning %in% c(
    'simvastatin', 'fluvastatin', 'pravastatin',
    'atorvastatin', 'rosuvastatin', 'lipid-lowering drug',
    'lipitor 10mg tablet'))
``` 

```{r read_statin_intake}
ukb_statin <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_meds.csv'),
  col_types = cols()) |>
  select(-med_other) |>
  pivot_longer(-eid, names_to = 'med', values_to = 'id') |>
  filter(!is.na(id)) |>
  inner_join(ukb_statin_codes, by = c('id' = 'coding')) |>
  select(eid) |>
  distinct() |>
  pull(eid)
```

### Construct Blood Biomarker covariates

```{r extract_blood_biomarker_names}
ukb_all_biochem_names <- ukb_all_biochem |> select(-eid) |> colnames()
```

```{r construct_blood_biomarker_covars_function}
construct_biochem_covars <- function(x, blood = 'Yes', statin = F, 
                                     scaling = T, retain_eid = F,
                                     crp = F, seq_centre = '') {
  
  if(blood == 'Yes') {
    aux <- ukb_gwas_covars |>
                     select(-Height)
  } else {
    aux <- ukb_gwas_covars |>
                     select(-Height, -starts_with('Blood'))
  }
  
  if(seq_centre != '') {
    aux <- aux |>
      filter(`Seq Centre` == seq_centre) |>
      select(-`Seq Centre`)
  }
  
  if (crp) 
    aux <- aux |> inner_join(
      ukb_all_biochem |> select(eid, `C-reactive protein`),
      by = 'eid')
  
  if (statin) {
    aux2 <- ukb_all_biochem |> filter(! eid %in% ukb_statin)
  } else {
    aux2 <- ukb_all_biochem
  }
  
  out <- aux2 |>
      select(eid, all_of(x)) |>
      inner_join(ukb_blood_extr_cov, by = 'eid') |>
      inner_join(ukb_biochem_proc_cov |>
                   select(eid,
                          all_of(paste(x, 'assay date')),
                          contains('dilution')),
                   by = 'eid') |>
      inner_join(ukb_fasting, by = 'eid') |>    
      inner_join(aux, by = 'eid') 
  
  if(! retain_eid) out <- out |> select(-eid) 
  
  if(scaling) {
    out |>
      mutate(across(where(is.numeric), scale))
  } else {
    out
  }
  
}
```

### Urine

```{r read_urine_biomarker_ids}
ukb_urine_ids <- read_csv(
  file.path(root_fd, 'data', 'releases', 'ukb_ids_urine.csv'),
  col_types = cols())

ukb_urine_bm_ids <- ukb_urine_ids |>
  filter(category == 'Urine assays',
         ! grepl('result flag', title))
```

```{r read_urine_biomarkers}
ukb_urine <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_urine.csv'),
  col_types = cols()) |> 
  group_by(eid) |>
  mutate_at(vars(-group_cols()), as.character) |>
  pivot_longer(-eid, names_to = 'id', values_to = 'val') |>
  group_by(id) |>
  separate(id, into = 'id', sep = '-', extra = 'drop') |>
  mutate(id = as.numeric(id)) |>
  inner_join(ukb_urine_bm_ids |> select(id, title), by = 'id') |>
  arrange(eid, title) |>
  select(-id) |>
  pivot_wider(names_from = title, values_from = val) |>
  mutate_all(as.numeric)
```

### Urine extraction date

```{r read_urine_extraction_dates}
ukb_urine_extr <- read_csv(
  file.path(root_fd, 'data', 'releases', 'pheno_12.csv'),
         col_types = cols()) |>
  select(eid, Extraction = `20035-0.0`) |>
  filter(!is.na(Extraction)) |>
  mutate(year = lubridate::year(Extraction)) |>
  filter(as.numeric(as.character(year)) > 1900) |>
  select(-year)
```

```{r construct_urine_extraction_covars}
ukb_urine_extr_cov <- ukb_blood_extr |> 
  mutate(time_hour = as.numeric(lubridate::hour(Extraction)),
         time_min = as.numeric(lubridate::minute(Extraction)),
         time_sec = as.numeric(lubridate::second(Extraction)),
         `Extr. Time` = time_hour + time_min/60 + time_sec/3600) |>
  select(-starts_with('time_')) |>
  mutate(`Extr. Month` = case_when(
    lubridate::year(Extraction) == 2006 ~ '2006',
    Extraction >= lubridate::as_date('2010-08-01') ~ '2010_end',
    TRUE ~ paste(lubridate::year(Extraction),
                 lubridate::month(Extraction), sep = '_'))) |>
  select(-Extraction)
```

### Urine acquisition time

```{r read_urine_processing_ids}
ukb_urine_proc_ids <- ukb_urine_ids |>
  filter(category == 'Urine processing')
```

```{r read_urine_processing}
ukb_urine_proc <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_urine.csv'),
  col_types = cols()) |>
  select(1, 10:17) |>
  rename_with(~ gsub("-0.0", "", .x, fixed = TRUE)) |>
  rename_with(~ ukb_urine_proc_ids |>
                filter(id == as.numeric(.x)) |>
                pull(title), -eid)
```

```{r construct_urine_processing_covars}
ukb_urine_proc_cov <- ukb_urine_proc |>
  select(eid, ends_with('acquisition time')) |>
  mutate(across(where(lubridate::is.Date),
                ~ as.numeric(.x - min(.x, na.rm = TRUE))))
```

### Construct Urine covariates

```{r find_urine_names}
ukb_urine_names <- ukb_urine |> select(-eid) |> colnames()
```

```{r construct_urine_biomarker_covars_function}
construct_urine_covars <- function(x, statin = F, seq_centre = '') {
  
  aux <- ukb_gwas_covars |>
    select(-Height, -starts_with('Blood'))
  
  if(seq_centre != '') {
    aux <- aux |>
      filter(`Seq Centre` == seq_centre) |>
      select(-`Seq Centre`)
  }
  
  if (statin) {
    aux2 <- ukb_urine |> filter(! eid %in% ukb_statin)
  } else {
    aux2 <- ukb_urine
  }
  
  aux2 |>
      select(eid, all_of(x)) |>
      inner_join(ukb_urine_extr_cov, by = 'eid') |>
      inner_join(ukb_urine_proc_cov |>
                   select(eid,
                          all_of(paste(x, 'acquisition time'))),
                   by = 'eid') |>
      inner_join(ukb_fasting, by = 'eid') |>    
      inner_join(aux, by = 'eid') |>
      select(-eid) |>
      mutate(across(where(is.numeric), scale))
  
}
```

## A.- Blood and Urine Biomarker Associations, excluding statin intake

### Blood

```{r fit_blood_biomarker_excl_statin_intake}
ukb_blood_biomarker_no_statin_sig <- bind_rows(
    lapply(
      X = c('Yes', 'No'),
      FUN = function(y) {
        lapply(
          X = ukb_all_biochem_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_biochem_covars(x, blood = y, statin = T)), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x, `Blood PCs` = y)
    }) } ))
```

### Urine

```{r fit_urine_biomarker_excl_statin_intake}
ukb_urine_biomarker_no_statin_sig <- bind_rows(
        lapply(
          X = ukb_urine_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_urine_covars(x, statin = T)), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x)
    }))
```

### Joint

```{r join_biomarker_associations_excl_statin_intake}
ukb_biomarkers_no_statin_sig <- bind_rows(
  ukb_blood_biomarker_no_statin_sig |> 
    mutate(type = if_else(`Blood PCs` == 'Yes',
                          'Blood Biomarker, with Blood PCs cov.',
                          'Blood Biomarker, w/o Blood PCs cov.'),
           tissue = 'Blood'),
  ukb_urine_biomarker_no_statin_sig |> 
    mutate(
      type = 'Urine Biomarker',
      tissue = 'Urine')) |>
  mutate(type = fct_rev(type))
```

```{r plot_biomarker_excl_statin_betas}
ukb_biomarkers_no_statin_betas_p <- ukb_biomarkers_no_statin_sig |>
  ggplot(aes(y = fct_rev(marker), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = type)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = type),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.03, 0.03),
                     breaks = c(-0.03, -0, 0.03)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  scale_shape_manual('',
                     values = c(
                       'Blood Biomarker, w/o Blood PCs cov.' = 21,
                      'Blood Biomarker, with Blood PCs cov.' = 23,
                      'Urine Biomarker' = 22)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = c(0.2, -0.075),
        legend.direction = 'horizontal',
        plot.margin = grid::unit(c(0.1,0.1,3,0.1), 'lines'))
```

```{r save_biomarker_excl_statin_betas_plot}
ggsave(ukb_biomarkers_no_statin_betas_p, 
       filename = '../figures/4.A.- Biomarkers Betas, no Statin.png',
       height = 12, width = 8)
```

```{r plot_biomarker_excl_statin_sig}
ukb_biomarkers_no_statin_sig_p <- ukb_biomarkers_no_statin_sig |>
  ggplot(aes(y = fct_rev(marker), x = lpval, group = type)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5, colour = 'grey80',
             linetype = 'dashed') +
  geom_col(aes(fill = type), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(0, 12.5),
                     breaks = c(0, 5, 10)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = c(0.25, -0.075),
        legend.direction = 'horizontal',
        plot.margin = grid::unit(c(0.1,0.1,3,0.1), 'lines'))
```

```{r save_biomarker_excl_statin_plots}
ggsave(
  plot = (ukb_biomarkers_no_statin_betas_p | (
    ukb_biomarkers_no_statin_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures', '4.A.- Biomarkers, no Statin.png'), 
  height = 11, width = 8)
```

## S13.- Blood and Urine Biomarker Associations, including statin intake

### Blood

```{r fit_blood_biomarker_incl_statin_intake}
ukb_blood_biomarker_sig <- bind_rows(
    lapply(
      X = c('Yes', 'No'),
      FUN = function(y) {
        lapply(
          X = ukb_all_biochem_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_biochem_covars(x, blood = y)), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x, `Blood PCs` = y)
    }) } ))
```

### Urine

```{r fit_urine_biomarker_incl_statin_intake}
ukb_urine_biomarker_sig <- bind_rows(
        lapply(
          X = ukb_urine_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_urine_covars(x)), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x)
    }))
```

### Joint

```{r join_biomarker_associations_incl_statin_intake}
ukb_biomarkers_sig <- bind_rows(
  ukb_blood_biomarker_sig |> 
    mutate(type = if_else(`Blood PCs` == 'Yes',
                          'Blood Biomarker, with Blood PCs cov.',
                          'Blood Biomarker, w/o Blood PCs cov.'),
           tissue = 'Blood'),
  ukb_urine_biomarker_sig |> 
    mutate(
      type = 'Urine Biomarker',
      tissue = 'Urine')) |>
  mutate(type = fct_rev(type))
```

```{r plot_biomarker_incl_statin_betas}
ukb_biomarkers_betas_p <- ukb_biomarkers_sig |>
  ggplot(aes(y = fct_rev(marker), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = type)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = type),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.03, 0.03),
                     breaks = c(-0.03, -0, 0.03)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  scale_shape_manual('',
                     values = c(
                       'Blood Biomarker, w/o Blood PCs cov.' = 21,
                      'Blood Biomarker, with Blood PCs cov.' = 23,
                      'Urine Biomarker' = 22)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = c(0.25, -0.075),
        legend.direction = 'horizontal',
        plot.margin = grid::unit(c(0.1,0.1,3,0.1), 'lines'))
```

```{r save_biomarker_incl_statin_betas_plot}
ggsave(ukb_biomarkers_betas_p, 
       filename = file.path(
         '..', 'figures_supp', 
         'S14.- Biomarkers Betas, including Statin.png'),
       height = 12, width = 10)
```

```{r plot_biomarker_incl_statin_sig}
ukb_biomarkers_sig_p <- ukb_biomarkers_sig |>
  ggplot(aes(y = fct_rev(marker), x = lpval, group = type)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5, colour = 'grey80',
             linetype = 'dashed') +
  geom_col(aes(fill = type), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = c(0.25, -0.075),
        legend.direction = 'horizontal',
        plot.margin = grid::unit(c(0.1,0.1,3,0.1), 'lines'))
```

```{r save_biomarker_incl_statin_plots}
ggsave(
  plot = (ukb_biomarkers_betas_p | (
    ukb_biomarkers_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures_supp', 
                       'S14.- Biomarkers, including Statin.png'), 
  height = 12, width = 10)
```

## S14i.- Blood and Urine Biomarker Associations, Sanger

### Blood

```{r fit_blood_biomarker_sanger}
ukb_blood_biomarker_sanger_sig <- bind_rows(
    lapply(
      X = c('Yes', 'No'),
      FUN = function(y) {
        lapply(
          X = ukb_all_biochem_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_biochem_covars(x, blood = y, statin = T, 
                                          seq_centre = 'Sanger')), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x, `Blood PCs` = y)
    }) } ))
```

### Urine

```{r fit_urine_biomarker_sanger}
ukb_urine_biomarker_sanger_sig <- bind_rows(
        lapply(
          X = ukb_urine_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_urine_covars(x, statin = T,
                                        seq_centre = 'Sanger')), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x)
    }))
```

### Joint

```{r join_biomarker_associations_sanger}
ukb_biomarkers_sanger_sig <- bind_rows(
  ukb_blood_biomarker_sanger_sig |> 
    mutate(type = if_else(`Blood PCs` == 'Yes',
                          'Blood Biomarker, with Blood PCs cov.',
                          'Blood Biomarker, w/o Blood PCs cov.'),
           tissue = 'Blood'),
  ukb_urine_biomarker_sanger_sig |> 
    mutate(
      type = 'Urine Biomarker',
      tissue = 'Urine')) |>
  mutate(type = fct_rev(type))
```

```{r plot_biomarker_sanger_betas}
ukb_biomarkers_sanger_betas_p <- ukb_biomarkers_sanger_sig |>
  ggplot(aes(y = fct_rev(marker), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = type)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = type),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.037, 0.037),
                     breaks = c(-0.03, -0, 0.03)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  scale_shape_manual('',
                     values = c(
                       'Blood Biomarker, w/o Blood PCs cov.' = 21,
                      'Blood Biomarker, with Blood PCs cov.' = 23,
                      'Urine Biomarker' = 22)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = 'none')
```

```{r plot_biomarker_sanger_sig}
ukb_biomarkers_sanger_sig_p <- ukb_biomarkers_sanger_sig |>
  ggplot(aes(y = fct_rev(marker), x = lpval, group = type)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5, colour = 'grey80',
             linetype = 'dashed') +
  geom_col(aes(fill = type), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(0, 7),
                     breaks = c(0, 3, 6)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = 'none')
```

```{r save_biomarker_sanger_plots}
ggsave(
  plot = (ukb_biomarkers_sanger_betas_p  | (
    ukb_biomarkers_sanger_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_annotation(title = '(i) Sanger') +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures_supp', 
                       'S15i.- Biomarkers, no Statin, Sanger.png'), 
  height = 11, width = 7)
```

## S14ii.- Blood and Urine Biomarker Associations, deCODE

### Blood

```{r fit_blood_biomarker_decode}
ukb_blood_biomarker_decode_sig <- bind_rows(
    lapply(
      X = c('Yes', 'No'),
      FUN = function(y) {
        lapply(
          X = ukb_all_biochem_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_biochem_covars(x, blood = y, statin = T, 
                                          seq_centre = 'deCODE')), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x, `Blood PCs` = y)
    }) } ))
```

### Urine

```{r fit_urine_biomarker_decode_intake}
ukb_urine_biomarker_decode_sig <- bind_rows(
        lapply(
          X = ukb_urine_names,
          FUN = function(x) {
            broom::tidy(
              lm(as.formula(paste0('`', x, '` ~ .')),
                 construct_urine_covars(x, statin = T,
                                        seq_centre = 'deCODE')), 
              conf.int = TRUE) |>
              mutate(lpval = -log10(p.value),
                     scoeff = sign(estimate)) |>
              filter(grepl('CN', term)) |>
              mutate(marker = x)
    }))
```

### Joint

```{r join_biomarker_associations_decode}
ukb_biomarkers_decode_sig <- bind_rows(
  ukb_blood_biomarker_decode_sig |> 
    mutate(type = if_else(`Blood PCs` == 'Yes',
                          'Blood Biomarker, with Blood PCs cov.',
                          'Blood Biomarker, w/o Blood PCs cov.'),
           tissue = 'Blood'),
  ukb_urine_biomarker_decode_sig |> 
    mutate(
      type = 'Urine Biomarker',
      tissue = 'Urine')) |>
  mutate(type = fct_rev(type))
```

```{r plot_biomarker_decode_betas}
ukb_biomarkers_decode_betas_p <- ukb_biomarkers_decode_sig |>
  ggplot(aes(y = fct_rev(marker), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = type)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = type),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.037, 0.037),
                     breaks = c(-0.03, -0, 0.03)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  scale_shape_manual('',
                     values = c(
                       'Blood Biomarker, w/o Blood PCs cov.' = 21,
                      'Blood Biomarker, with Blood PCs cov.' = 23,
                      'Urine Biomarker' = 22)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = 'none')
```

```{r plot_biomarker_decode_sig}
ukb_biomarkers_decode_sig_p <- ukb_biomarkers_decode_sig |>
  ggplot(aes(y = fct_rev(marker), x = lpval, group = type)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5, colour = 'grey80',
             linetype = 'dashed') +
  geom_col(aes(fill = type), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  facet_grid(rows = vars(tissue), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(0, 7),
                     breaks = c(0, 3, 6)) +
  scale_fill_manual('', 
                    values = c(
                      'Blood Biomarker, w/o Blood PCs cov.' = 'coral4',
                      'Blood Biomarker, with Blood PCs cov.' = 'mistyrose2',
                      'Urine Biomarker' = 'goldenrod3')) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.text = element_text(size = 11),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12))  +
  theme(legend.position = 'none')
```

```{r save_biomarker_decode_plots}
ggsave(
  plot = (ukb_biomarkers_decode_betas_p | (
    ukb_biomarkers_decode_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))) +
  plot_annotation(title = '(ii) deCODE') +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures_supp', 
                       'S15ii.- Biomarkers, no Statin, deCODE.png'), 
  height = 11, width = 7)
```


## S15A.- Multivariate Analysis of Renal Blood Biomarkers 

```{r define_renal_biomarkers}
renal_biom <- c('Cystatin C', 'Creatinine', 'Urea', 'Urate', 
                'Phosphate', 'Total protein')
```

### No CRP as covariate

```{r construct_renal_biomarkers_covars}
ukb_renal_biom_cov <- construct_biochem_covars('Cystatin C', 
                         blood = 'No', scaling = T) |>
    bind_cols(lapply(
      X = renal_biom[-1],
      FUN = function(x) {
        construct_biochem_covars(x,  blood = 'No', scaling = T) |>
                   select(contains(x)) })) |>
  select(-contains('assay date'))
```

```{r fit_renal_biomarkers}
ukb_renal_biom_sig <- broom::tidy(
  lm(CN ~ . , ukb_renal_biom_cov), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term %in% renal_biom)
```

### CRP as covariate

```{r construct_renal_biomarkers_covars_with_crp}
ukb_renal_biom_crp_cov <- construct_biochem_covars('Cystatin C', 
                         blood = 'No', scaling = T, crp = T) |>
    bind_cols(lapply(
      X = renal_biom[-1],
      FUN = function(x) {
        construct_biochem_covars(x,  blood = 'No', 
                                 scaling = T, crp = T) |>
                   select(contains(x)) })) |>
  select(-contains('assay date'))
```

```{r fit_renal_biomarkers_with_crp}
ukb_renal_biom_crp_sig <- broom::tidy(
  lm(CN ~ . , ukb_renal_biom_crp_cov), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term %in% renal_biom)
```

### Joint Plot

```{r join_renal_biomarkers_associations}
ukb_renal_joint_sig <- bind_rows(
  ukb_renal_biom_crp_sig |>
    mutate(type = 'Renal Function\nBlood Biomarkers',
           covariates = 'Yes'),
  ukb_renal_biom_sig |>
    mutate(type = 'Renal Function\nBlood Biomarkers',
           covariates = 'No')) |>
  mutate(covariates = covariates)
```

```{r plot_renal_biomarkers_betas}
ukb_renal_joint_betas_p <- ukb_renal_joint_sig |>
  ggplot(aes(y = fct_rev(term), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = covariates, group = covariates)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = covariates),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(type), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.03, 0.03),
                     breaks = c(-0.03, 0, 0.03)) +
  scale_fill_manual('CRP as cov.',
                    values = c('No' = 'chocolate4',
                               'Yes' = 'bisque3')) +
  scale_shape_manual('CRP as cov.',
                     values = c(
                       'No' = 21,
                       'Yes' = 24)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.direction = 'horizontal',
        legend.title = element_text(face = 'bold'),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12),
        plot.margin = grid::unit(c(0.1,0.1,2,0.1), 'lines'))
```

```{r plot_renal_biomarkers_sig}
ukb_renal_joint_sig_p <- ukb_renal_joint_sig |>
  ggplot(aes(y = fct_rev(term), 
             x = lpval,
             group = covariates)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5,
             colour = 'grey80', linetype = 'dashed') +
  geom_col(aes(fill = covariates), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  facet_grid(rows = vars(type), scales = 'free', space = 'free') +
  scale_fill_manual('CRP as cov.',
                    values = c('No' = 'chocolate4',
                               'Yes' = 'bisque3')) +
  scale_shape_manual('CRP as cov.',
                     values = c(
                       'No' = 21,
                       'Yes' = 24)) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.direction = 'horizontal',
        legend.title = element_text(face = 'bold'),
        strip.text = element_blank(),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.y = element_text(size = 12),
        plot.margin = grid::unit(c(0.1,0.1,2,0.1), 'lines'))
```

```{r save_renal_biomarkers_plots}
ggsave(
  plot = (ukb_renal_joint_betas_p | (
    ukb_renal_joint_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures_supp',
                       'S16A.- Renal Biomarkers, Joint.png'), 
  height = 5.5, width = 10)
```

## eGFR associations

### No CRP as covariate

#### eGFR from creatinine

```{r construct_egfr_cr_covars}
ukb_egfr_cr <- construct_biochem_covars('Creatinine',
                                        blood = 'No', scaling = F) |>
  filter(!is.na(Creatinine)) |>
  mutate(
    gamma = if_else(Sex == 'Male', 142, 143.704),
    kappa = if_else(Sex == 'Male', 0.9, 0.7),
    alpha = if_else(Sex == 'Male', -.302, -.241)) |>
  rowwise() |>
  mutate(
    eGFRcr = gamma *
      min(Creatinine*0.0113/kappa, 1)^alpha *
      max(Creatinine*0.0113/kappa, 1)^-1.2 *
      0.9938^Age) |>
  ungroup()
```

```{r scale_egfr_cr_covars}
ukb_egfr_cr_scaled <- ukb_egfr_cr |>
  select(-alpha, -kappa, -gamma, -Creatinine) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr}
ukb_egfr_cr_sig <- broom::tidy(
  lm( eGFRcr ~ . , ukb_egfr_cr_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr')
```

#### GFR from  Cystatin C

```{r construct_egfr_cc_covars}
ukb_egfr_cc <- construct_biochem_covars(
  'Cystatin C',  blood = 'No', scaling = F) |>
  filter(!is.na(`Cystatin C`)) |>
  mutate(
    gamma = if_else(Sex == 'Male', 133, 133*0.932)) |>
  rowwise() |>
  mutate(
    eGFRcc = gamma *
      min(`Cystatin C`/0.8, 1)^-0.499 *
      max(`Cystatin C`/0.8, 1)^-1.328 *
      0.996^Age) |>
  ungroup()
```

```{r scale_egfr_cc_covars}
ukb_egfr_cc_scaled <- ukb_egfr_cc |>
  select(-gamma, -`Cystatin C`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cc}
ukb_egfr_cc_sig <- broom::tidy(
  lm(eGFRcc ~ . , ukb_egfr_cc_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcc')
```

#### GFR from Creatinine + Cystatin C

```{r construct_egfr_cr_cc_covars}
ukb_egfr_cr_cc <- construct_biochem_covars('Cystatin C', 
                         blood = 'No', scaling = F,
                         retain_eid = TRUE) |>
    bind_cols(construct_biochem_covars(
      'Creatinine',  blood = 'No', scaling = F) |>
                   select(contains('Creatinine'))) |>
    filter(!is.na(Creatinine), !is.na(`Cystatin C`)) |>
    select(-contains('assay date'))  |>
  mutate(
    gamma = if_else(Sex == 'Male', 135, 135*0.963),
    kappa = if_else(Sex == 'Male', 0.9, 0.7),
    alpha = if_else(Sex == 'Male', -.144, -.219)) |>
  rowwise() |>
  mutate(
    `eGFRcr+cc` = gamma *
      min(Creatinine*0.0113/kappa, 1)^alpha *
      max(Creatinine*0.0113/kappa, 1)^-0.544 *
      min(`Cystatin C`/0.8, 1)^-0.323 *
      max(`Cystatin C`/0.8, 1)^-0.778 *
      0.9961^Age) |>
  ungroup()
```

```{r clean_egfr_cr_cc_covars}
ukb_egfr_cr_cc_cov <- ukb_egfr_cr_cc |>
  select(-alpha, -kappa, -gamma, -Creatinine, -`Cystatin C`)
```

```{r save_egfr_cr_cc_covars}
ukb_egfr_cr_cc_cov |>
  write_csv(file.path(root_fd, 'data', 'pheno', 'egfr_cr_cc_cov.csv'),
            na = '')
```

```{r scale_egfr_cr_cc_covars}
ukb_egfr_cr_cc_scaled <- ukb_egfr_cr_cc_cov |>
  select(-eid) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_cc}
ukb_egfr_cr_cc_sig <- broom::tidy(
  lm(`eGFRcr+cc` ~ . , ukb_egfr_cr_cc_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr+cc')
```

#### Join eGFR associations without CRP

```{r join_egfr_associations_without_crp}
ukb_egfr_sig <- bind_rows(
  ukb_egfr_cr_sig,
  ukb_egfr_cc_sig,
  ukb_egfr_cr_cc_sig
)
```

### CRP as covariate

#### GFR from creatinine

```{r construct_egfr_cr_covars_with_crp}
ukb_egfr_cr_crp <- construct_biochem_covars(
  'Creatinine', blood = 'No', scaling = F, crp = T) |>
  filter(!is.na(Creatinine)) |>
  mutate(
    gamma = if_else(Sex == 'Male', 142, 143.704),
    kappa = if_else(Sex == 'Male', 0.9, 0.7),
    alpha = if_else(Sex == 'Male', -.302, -.241)) |>
  rowwise() |>
  mutate(
    eGFRcr = gamma *
      min(Creatinine*0.0113/kappa, 1)^alpha *
      max(Creatinine*0.0113/kappa, 1)^-1.2 *
      0.9938^Age) |>
  ungroup()
```

```{r scale_egfr_cr_covars_with_crp}
ukb_egfr_cr_crp_scaled <- ukb_egfr_cr_crp |>
  select(-alpha, -kappa, -gamma, -Creatinine) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_with_crp}
ukb_egfr_cr_crp_sig <- broom::tidy(
  lm(eGFRcr ~ . , ukb_egfr_cr_crp_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr')
```

#### GFR from  Cystatin C

```{r construct_egfr_cc_covars_with_crp}
ukb_egfr_cc_crp <- construct_biochem_covars(
  'Cystatin C', blood = 'No', scaling = F, crp = T) |>
  filter(!is.na(`Cystatin C`)) |>
  mutate(
    gamma = if_else(Sex == 'Male', 133, 133*0.932)) |>
  rowwise() |>
  mutate(
    eGFRcc = gamma *
      min(`Cystatin C`/0.8, 1)^-0.499 *
      max(`Cystatin C`/0.8, 1)^-1.328 *
      0.996^Age) |>
  ungroup()
```

```{r scale_egfr_cc_covars_with_crp}
ukb_egfr_cc_crp_scaled <- ukb_egfr_cc_crp |>
  select(-gamma, -`Cystatin C`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cc_with_crp}
ukb_egfr_cc_crp_sig <- broom::tidy(
  lm(eGFRcc ~ . , ukb_egfr_cc_crp_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcc')
```

#### GFR from Creatinine + Cystatin C

```{r construct_egfr_cr_cc_covars_with_crp}
ukb_egfr_cr_cc_crp <- construct_biochem_covars(
  'Cystatin C', blood = 'No', scaling = F, crp = T,
  retain_eid = TRUE) |>
    bind_cols(construct_biochem_covars(
      'Creatinine', blood = 'No', scaling = F, crp = T) |>
        select(contains('Creatinine'))) |>
    filter(!is.na(Creatinine), !is.na(`Cystatin C`)) |>
    select(-contains('assay date'))  |>
  mutate(
    gamma = if_else(Sex == 'Male', 135, 135*0.963),
    kappa = if_else(Sex == 'Male', 0.9, 0.7),
    alpha = if_else(Sex == 'Male', -.144, -.219)) |>
  rowwise() |>
  mutate(
    `eGFRcr+cc` = gamma *
      min(Creatinine*0.0113/kappa, 1)^alpha *
      max(Creatinine*0.0113/kappa, 1)^-0.544 *
      min(`Cystatin C`/0.8, 1)^-0.323 *
      max(`Cystatin C`/0.8, 1)^-0.778 *
      0.9961^Age) |>
  ungroup()
```

```{r scale_egfr_cr_cc_covars_with_crp}
ukb_egfr_cr_cc_crp_scaled <- ukb_egfr_cr_cc_crp |>
  select(-eid, -alpha, -kappa, -gamma, -Creatinine, -`Cystatin C`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_cc_with_crp}
ukb_egfr_cr_cc_crp_sig <- broom::tidy(
  lm(`eGFRcr+cc` ~ . , ukb_egfr_cr_cc_crp_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr+cc')
```

#### Join eGFRs associations with CRP

```{r join_egfr_associations_with_crp}
ukb_egfr_crp_sig <- bind_rows(
  ukb_egfr_cr_crp_sig,
  ukb_egfr_cc_crp_sig,
  ukb_egfr_cr_cc_crp_sig)
```

### Plot eGFR associations

```{r join_egfr_associations}
egfr_labels <- ukb_egfr_crp_sig |>
  arrange(desc(term)) |>
  mutate(term = gsub('R', 'R<sub>', gsub('$', '</sub>', term))) |>
  pull(term)
ukb_egfr_all_sig <- bind_rows(
  ukb_egfr_crp_sig |>
    mutate(type = 'GFR\nEstimates',
           covariates = 'Yes'),
  ukb_egfr_sig |>
    mutate(type = 'GFR\nEstimates',
           covariates = 'No'))
```

```{r plot_egfr_betas}
ukb_egfr_betas_p <- ukb_egfr_all_sig |>
  ggplot(aes(y = forcats::fct_rev(term), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = covariates, group = covariates)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = covariates),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(type), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.031, 0.031),
                     breaks = c(-0.03, 0, 0.03)) +
  scale_y_discrete(labels = egfr_labels) +
  scale_fill_manual('CRP as cov.',
                    values = c('No' = 'chocolate4',
                               'Yes' = 'bisque3')) +
  scale_shape_manual('CRP as cov.',
                     values = c(
                       'No' = 21,
                       'Yes' = 24)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.direction = 'horizontal',
        legend.title = element_text(face = 'bold'),
        strip.text = element_blank(),
        axis.text.y = ggtext::element_markdown(size = 12),
        axis.title = element_text(size = 14),
        plot.margin = grid::unit(c(0.1,0.1,2,0.1), 'lines'))
```

```{r save_egfr_betas_plot}
ggsave(ukb_egfr_betas_p, 
       filename = '../figures/4.B.- eGFR Betas.png',
       height = 3, width = 8)
```

```{r plot_egfr_sig}
ukb_egfr_sig_p <- ukb_egfr_all_sig |>
  ggplot(aes(y = fct_rev(term), x = lpval, group = covariates)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5, colour = 'grey80',
             linetype = 'dashed') +
  geom_col(aes(fill = covariates), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  scale_y_discrete(labels = egfr_labels) +
  scale_fill_manual('CRP as cov.',
                    values = c('No' = 'chocolate4',
                               'Yes' = 'bisque3')) +
  scale_shape_manual('CRP as cov.',
                     values = c(
                       'No' = 21,
                       'Yes' = 24)) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.direction = 'horizontal',
        legend.title = element_text(face = 'bold'),
        strip.text = element_blank(),
        axis.text.y = ggtext::element_markdown(size = 12),
        axis.title = element_text(size = 14),
        plot.margin = grid::unit(c(0.1,0.1,2,0.1), 'lines'))
```

```{r save_egfr_plots}
ggsave(
  plot = (ukb_egfr_betas_p | (
    ukb_egfr_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures', '4.B.- eGFR.png'), 
  height = 4, width = 6)
```

## eGFR by Sequencing centre

### Sanger

#### No CRP as covariate

```{r scale_egfr_cr_covars_sanger}
ukb_egfr_cr_sanger_scaled <- ukb_egfr_cr |>
  filter(`Seq Centre` == 'Sanger') |>
  select(-alpha, -kappa, -gamma, -Creatinine, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_sanger}
ukb_egfr_cr_sanger_sig <- broom::tidy(
  lm( eGFRcr ~ . , ukb_egfr_cr_sanger_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr')
```

```{r scale_egfr_cc_sanger}
ukb_egfr_cc_sanger_scaled <- ukb_egfr_cc |>
  filter(`Seq Centre` == 'Sanger') |>
  select(-gamma, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cc_sanger}
ukb_egfr_cc_sanger_sig <- broom::tidy(
  lm(eGFRcc ~ . , ukb_egfr_cc_sanger_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcc')
```

```{r scale_egfr_cr_cc_sanger}
ukb_egfr_cr_cc_sanger_scaled <- ukb_egfr_cr_cc |>
  filter(`Seq Centre` == 'Sanger') |>
  select(-eid, -alpha, -kappa, -gamma, 
         -Creatinine, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_cc_sanger}
ukb_egfr_cr_cc_sanger_sig <- broom::tidy(
  lm(`eGFRcr+cc` ~ . , ukb_egfr_cr_cc_sanger_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr+cc')
```

```{r join_egfr_associations_sanger_without_crp}
ukb_egfr_sanger_sig <- bind_rows(
  ukb_egfr_cr_sanger_sig,
  ukb_egfr_cc_sanger_sig,
  ukb_egfr_cr_cc_sanger_sig)
```

#### CRP as covariate

```{r scale_egfr_cr_sanger_with_crp}
ukb_egfr_cr_crp_sanger_scaled <- ukb_egfr_cr_crp |>
  filter(`Seq Centre` == 'Sanger') |>
  select(-alpha, -kappa, -gamma, -Creatinine, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_sanger_with_crp}
ukb_egfr_cr_crp_sanger_sig <- broom::tidy(
  lm(eGFRcr ~ . , ukb_egfr_cr_crp_sanger_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr')
```

```{r scale_egfr_cc_sanger_with_crp}
ukb_egfr_cc_crp_sanger_scaled <- ukb_egfr_cc_crp |>
  filter(`Seq Centre` == 'Sanger') |>
  select(-gamma, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cc_sanger_with_crp}
ukb_egfr_cc_crp_sanger_sig <- broom::tidy(
  lm(eGFRcc ~ . , ukb_egfr_cc_crp_sanger_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcc')
```

```{r scale_egfr_cr_cc_sanger_with_crp}
ukb_egfr_cr_cc_crp_sanger_scaled <- ukb_egfr_cr_cc_crp |>
  filter(`Seq Centre` == 'Sanger') |>
  select(-eid, -alpha, -kappa, -gamma,
         -Creatinine, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_cc_sanger_with_crp}
ukb_egfr_cr_cc_crp_sanger_sig <- broom::tidy(
  lm(`eGFRcr+cc` ~ . , ukb_egfr_cr_cc_crp_sanger_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr+cc')
```

```{r join_egfr_associations_sanger_with_crp}
ukb_egfr_crp_sanger_sig <- bind_rows(
  ukb_egfr_cr_crp_sanger_sig,
  ukb_egfr_cc_crp_sanger_sig,
  ukb_egfr_cr_cc_crp_sanger_sig)
```

### deCODE

#### No CRP as covariate

```{r scale_egfr_cr_decode}
ukb_egfr_cr_decode_scaled <- ukb_egfr_cr |>
  filter(`Seq Centre` == 'deCODE') |>
  select(-alpha, -kappa, -gamma, -Creatinine, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_decode}
ukb_egfr_cr_decode_sig <- broom::tidy(
  lm( eGFRcr ~ . , ukb_egfr_cr_decode_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr')
```

```{r scale_egfr_cc_decode}
ukb_egfr_cc_decode_scaled <- ukb_egfr_cc |>
  filter(`Seq Centre` == 'deCODE') |>
  select(-gamma, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cc_decode}
ukb_egfr_cc_decode_sig <- broom::tidy(
  lm(eGFRcc ~ . , ukb_egfr_cc_decode_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcc')
```

```{r scale_egfr_cr_cc_decode}
ukb_egfr_cr_cc_decode_scaled <- ukb_egfr_cr_cc |>
  filter(`Seq Centre` == 'deCODE') |>
  select(-eid, -alpha, -kappa, -gamma, 
         -Creatinine, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_cc_decode}
ukb_egfr_cr_cc_decode_sig <- broom::tidy(
  lm(`eGFRcr+cc` ~ . , ukb_egfr_cr_cc_decode_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr+cc')
```

```{r join_egfr_associations_decode_without_crp}
ukb_egfr_decode_sig <- bind_rows(
  ukb_egfr_cr_decode_sig,
  ukb_egfr_cc_decode_sig,
  ukb_egfr_cr_cc_decode_sig)
```

#### CRP as covariate

```{r scale_egfr_cr_decode_with_crp}
ukb_egfr_cr_crp_decode_scaled <- ukb_egfr_cr_crp |>
  filter(`Seq Centre` == 'deCODE') |>
  select(-alpha, -kappa, -gamma, -Creatinine, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_decode_with_crp}
ukb_egfr_cr_crp_decode_sig <- broom::tidy(
  lm(eGFRcr ~ . , ukb_egfr_cr_crp_decode_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr')
```

```{r scale_egfr_cc_decode_with_crp}
ukb_egfr_cc_crp_decode_scaled <- ukb_egfr_cc_crp |>
  filter(`Seq Centre` == 'deCODE') |>
  select(-gamma, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cc_decode_with_crp}
ukb_egfr_cc_crp_decode_sig <- broom::tidy(
  lm(eGFRcc ~ . , ukb_egfr_cc_crp_decode_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcc')
```

```{r scale_egfr_cr_cc_decode_with_crp}
ukb_egfr_cr_cc_crp_decode_scaled <- ukb_egfr_cr_cc_crp |>
  filter(`Seq Centre` == 'deCODE') |>
  select(-eid, -alpha, -kappa, -gamma,
         -Creatinine, -`Cystatin C`, -`Seq Centre`) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_egfr_cr_cc_decode_with_crp}
ukb_egfr_cr_cc_crp_decode_sig <- broom::tidy(
  lm(`eGFRcr+cc` ~ . , ukb_egfr_cr_cc_crp_decode_scaled), 
  conf.int = TRUE) |>
  mutate(lpval = -log10(p.value),
         scoeff = sign(estimate),
         term = gsub('`', '', term)) |>
  filter(term == 'CN') |>
  mutate(term = 'eGFRcr+cc')
```

```{r join_egfr_associations_decode_with_crp}
ukb_egfr_crp_decode_sig <- bind_rows(
  ukb_egfr_cr_crp_decode_sig,
  ukb_egfr_cc_crp_decode_sig,
  ukb_egfr_cr_cc_crp_decode_sig
)
```

### Combined

```{r join_egfr_associations_by_seq_centre}
ukb_egfr_sc_sig <- bind_rows(
  ukb_egfr_sanger_sig |> mutate(covariates = 'No', SC = 'Sanger'),
  ukb_egfr_crp_sanger_sig |> mutate(covariates = 'Yes', SC = 'Sanger'),
  ukb_egfr_decode_sig |> mutate(covariates = 'No', SC = 'deCODE'),
  ukb_egfr_crp_decode_sig |> mutate(covariates = 'Yes', SC = 'deCODE'))
```

```{r plot_egfr_seq_centre_betas}
ukb_egfr_sc_betas_p <- ukb_egfr_sc_sig |>
  ggplot(aes(y = forcats::fct_rev(term), x = estimate, 
             xmin = conf.low, xmax = conf.high,
             shape = covariates, group = covariates)) +
  geom_vline(xintercept = 0, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(aes(fill = covariates),
                  colour = 'grey30', size = 1,
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  facet_grid(rows = vars(SC), scales = 'free', space = 'free') +
  scale_x_continuous(limits = c(-0.03, 0.03),
                     breaks = c(-0.03, 0, 0.03)) +
  scale_y_discrete(labels = egfr_labels) +
  scale_fill_manual('CRP as cov.',
                    values = c('No' = 'chocolate4',
                               'Yes' = 'bisque3')) +
  scale_shape_manual('CRP as cov.',
                     values = c(
                       'No' = 21,
                       'Yes' = 24)) +
  labs(y = '', x = "Standardised Beta; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.direction = 'horizontal',
        legend.title = element_text(face = 'bold'),
        strip.text = element_blank(),
        axis.text.y = ggtext::element_markdown(size = 12),
        axis.title = element_text(size = 14),
        plot.margin = grid::unit(c(0.1,0.1,2,0.1), 'lines'))
```

```{r plot_egfr_seq_centre_sig}
ukb_egfr_sc_sig_p <- ukb_egfr_sc_sig |>
  ggplot(aes(y = fct_rev(term), x = lpval, group = covariates)) +
  geom_vline(xintercept = -log10(0.05), linewidth = 0.5, colour = 'grey80',
             linetype = 'dashed') +
  geom_col(aes(fill = covariates), colour = 'grey30', 
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.5)) +
  facet_grid(rows = vars(SC), scales = 'free', space = 'free') +
  scale_y_discrete(labels = egfr_labels) +
  scale_fill_manual('CRP as cov.',
                    values = c('No' = 'chocolate4',
                               'Yes' = 'bisque3')) +
  scale_shape_manual('CRP as cov.',
                     values = c(
                       'No' = 21,
                       'Yes' = 24)) +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.direction = 'horizontal',
        legend.title = element_text(face = 'bold'),
        axis.text.y = ggtext::element_markdown(size = 12),
        axis.title = element_text(size = 14),
        plot.margin = grid::unit(c(0.1,0.1,2,0.1), 'lines'))
```

```{r save_egfr_seq_centre_plots}
ggsave(
  plot = (ukb_egfr_sc_betas_p | (
    ukb_egfr_sc_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures_supp', 'S16B.- eGFR, SC.png'), 
  height = 6, width = 10)
```

## Staircase eGFR

```{r define_group_labels}
group_labels <- c('Low', 'Mid-Low', 'Mid', 'Mid-High', 'High')
```

```{r split_egfr_values}
ukb_egfr_cr_cc_split <- ukb_egfr_cr_cc |>
  mutate(eGFR_group = cut_number(`eGFRcr+cc`,
                                 n = 5, labels = group_labels))
```

```{r read_adj_cn}
ukb_adj_47s <- read_csv(
  file.path(root_fd, 'data', 'cn', 'ukb_adj_cn_wb.csv'),
  col_types = cols())
```

```{r slice_egfr_cn}
ukb_egfr_ci_adj <- ukb_egfr_cr_cc_split |>
    inner_join(ukb_adj_47s, by = 'eid') |>
    group_by(eGFR_group) |>
    summarise(value = mean(adj_CN),
              SD = sd(adj_CN),
              N = n_distinct(eid),
              margin = qt(ci,df=N-1)*SD/sqrt(N),
              value_low = value - margin,
              value_high = value + margin,
              .groups = 'drop')
```

```{r plot_egfr_groups}
ukb_egfr_ci_adj_p <- ukb_egfr_ci_adj |>
  ggplot(aes(y = eGFR_group, fill = eGFR_group,
             xmin = value_low, xmax = value_high, x = value)) +
  geom_crossbar(color = 'grey20', alpha = 0.8, linewidth = 0.75) +
  scale_x_continuous(limits = c(1.585e-4, 1.635e-4),
                     breaks = c(1.59e-4, 1.61e-4, 1.63e-4),
                     labels = fancy_scientific) +
  scale_fill_brewer(palette = 'Oranges') +
  labs(y = 'eGFR<sub>cr+cc</sub> Group', 
       x = bquote(bold(atop("SC-Adjusted 18S Ratio (Mean" %+-% "95% CI)")))) +
  theme_pub() +
  theme(legend.position = 'none',
        axis.title.y = element_markdown())
```

```{r save_egfr_groups_plot}
ggsave(ukb_egfr_ci_adj_p, 
       filename = '../figures/4.C.- Staircase eGFR.png',
       height = 4, width = 6)
```

## Kidney disease

```{r read_kidney_diseases}
ukb_kidney_d_ids <- read_csv(
  file.path(root_fd, 'data', 'releases', 'ukb_ids_kidney_d.csv'),
  col_types = cols())
```

```{r encode_kidney_diseases, warning = F}
ukb_kidney_d <- read_csv(
 file.path(root_fd, 'data', 'pheno', 'ukb_kidney_d.csv'),
  col_types = cols()) |> 
  group_by(eid) |>
  pivot_longer(-eid, names_to = 'id', values_to = 'val') |>
    mutate(val = if_else(
      as.numeric(lubridate::year(val)) < 1904 |
        as.numeric(lubridate::year(val)) > 2036, NA, val)) |>
  mutate(has = ! is.na(val)) |>
  select(-val) |>
  group_by(id) |>
  separate(id, into = 'id', sep = '-', extra = 'drop') |>
  mutate(id = as.numeric(id)) |>
  inner_join(ukb_kidney_d_ids |> select(id, title), by = 'id') |>
  mutate(title = gsub('Date |first reported ', '', title)) |>
  arrange(eid, title) |>
  select(-id) |>
  pivot_wider(names_from = title, values_from = has)
```

```{r define_kidney_disease_groups}
ukb_glom_d_c <- paste0('N0', 1:8)
ukb_fail_kidney_c <- paste0('N', c(17:18))
ukb_urolithiasis_c <- paste0('N', c(20:23))
```

```{r group_kidney_diseases}
ukb_group_kidney <- ukb_kidney_d |>
  pivot_longer(-eid) |>
  group_by(name) |>
  separate(name, into = c('code'), sep = ' ', extra = 'drop') |>
  filter(code %in% c(ukb_glom_d_c, ukb_fail_kidney_c, ukb_urolithiasis_c)) |>
  mutate(kidney_g = case_when(
    code %in% ukb_glom_d_c ~ 'Glomerular D.',
    code %in% ukb_fail_kidney_c ~ 'Kidney Fail.', 
    code %in% ukb_urolithiasis_c ~ 'Urolithiasis')) |>
  group_by(eid, kidney_g) |>
  summarise(`Kidney D.` = any(value), .groups = 'drop') |>
  pivot_wider(names_from = 'kidney_g', values_from = `Kidney D.`) |>
  ungroup()
```

```{r}
ukb_group_kidney |>
  write_csv(file.path(root_fd, 'data', 'pheno', 'ukb_kidney_d_groups.csv'),
            na = '')
```

```{r construct_kidney_disease_covars}
ukb_group_kidney_covar <- ukb_group_kidney |>
  inner_join(ukb_gwas_covars |>
               select(-Height, -starts_with('Blood')), by = 'eid')
```

```{r scale_kidney_disease_covars}
ukb_group_kidney_scaled <- ukb_group_kidney_covar |>
  select(-eid) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r define_kidney_disease_group_names}
kidney_gs <- colnames(ukb_group_kidney)[-1]
```

```{r fit_kidney_disease_function}
fit_kidney_logit <- function(df) {
  bind_rows(
    lapply(
      X = kidney_gs,
      FUN = function(x) {
        kidney_gs_other <- setdiff(kidney_gs, x)
        broom::tidy(
          glm(as.formula(paste0('`', stringr::str_trim(x), '` ~ . ')),
              df |>
               select(-all_of(kidney_gs_other)), 
          family = 'binomial'),
          conf.int = TRUE) |>
          mutate(lpval = -log10(p.value),
                 scoeff = sign(estimate)) |>
          filter(term == 'CN') |>
          select(-term) |>
          mutate(term = x) |>
          relocate(term)
      }))
}
```

```{r fit_kidney_diseases}
ukb_group_kidney_logit_sig <- fit_kidney_logit(ukb_group_kidney_scaled)
```

```{r calculate_kidney_diseases_odds_ratios}
ukb_group_kidney_or <-  ukb_group_kidney_logit_sig |>
  mutate(OR = exp(estimate), OR_l = exp(conf.low), OR_h = exp(conf.high)) |>
  mutate(
    name = case_when(
     grepl('Gl', term) ~ 'Glomerular\nDisease\n(N00-N08)\n[N = 1,310]', 
     grepl('Kid', term) ~ 'Kidney\nFailure\n(N17-N18)\n[N = 12,689]',
     grepl('Uro', term) ~ '\nUrolithiasis\n(N20-N23)\n[N = 5,144]'
    ) ) |>
  mutate(name = forcats::fct_rev(forcats::fct_inorder(name)))
```

```{r plot_kidney_disease_odds_ratios}
ukb_group_kidney_logit_beta_p <- ukb_group_kidney_or |>
  ggplot(aes(y = name, x = OR, xmin = OR_l, xmax = OR_h)) +
  geom_vline(xintercept = 1, linewidth = 0.5, colour = 'grey80') +
  geom_pointrange(colour = 'grey30', size = 1.5, shape = 22,
                  fill = 'sienna3',
                linewidth = 0.75, alpha = 0.8,
                position = position_dodge2(width = 0.75)) +
  scale_x_continuous(limits = c(0.925, 1.075),
                     breaks = c(0.95, 1, 1.05)) +
  labs(y = '', x = "Odds Ratio; 95% CI") +
  theme_pub() +
  theme(legend.position = 'bottom',
        strip.text = element_text(size = 12),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.x = element_text(size = 12))
```

```{r plot_kidney_diseases_sig}
ukb_group_kidney_logit_sig_p <- ukb_group_kidney_or |>
  ggplot(aes(y = name, x = lpval)) +
  geom_col(colour = 'grey30', size = 1, fill = 'grey85',
                linewidth = 0.75, alpha = 0.8, width = 0.65,
                position = position_dodge2(width = 0.65)) +
  geom_vline(xintercept = -log10(.05), 
             linewidth = 0.5, colour = 'grey70',
             linetype = 'dashed') +
  labs(y = '', x = bquote(bold(-log[10](p-value)))) +
  theme_pub() +
  theme(legend.position = 'bottom',
        strip.text = element_text(size = 12),
        axis.text = element_text(size = 12),
        axis.title = element_text(size = 14),
        axis.text.x = element_text(size = 12))
```

```{r save_kidney_diseases_plots}
ggsave(
  plot = (ukb_group_kidney_logit_beta_p | (
    ukb_group_kidney_logit_sig_p +
      theme(axis.text.y = element_blank(),
            axis.title.y = element_blank(),
            axis.ticks.y = element_blank()))
    ) +
  plot_layout(widths = c(0.6, 0.4)),
  filename = file.path('..', 'figures', '4.D.- Kidney Disease.png'), 
  height = 3.5, width = 6.5)
```

### Other Ethnicities

```{r read_kidney_diseases_all_eth, warning = F}
ukb_kidney_d_all_eth <- read_csv(
  paste0('~/OneDrive - Queen Mary, University of London/comp_bio/', 
         'human_rdna/ukb/data/pheno/ukb_kidney_d_all_eth.csv'),
  col_types = cols()) |> 
  group_by(eid) |>
  pivot_longer(-eid, names_to = 'id', values_to = 'val') |>
    mutate(val = if_else(
      as.numeric(lubridate::year(val)) < 1904 |
        as.numeric(lubridate::year(val)) > 2036, NA, val)) |>
  mutate(has = ! is.na(val)) |>
  select(-val) |>
  group_by(id) |>
  separate(id, into = 'id', sep = '-', extra = 'drop') |>
  mutate(id = as.numeric(id)) |>
  inner_join(ukb_kidney_d_ids |> select(id, title), by = 'id') |>
  mutate(title = gsub('Date |first reported ', '', title)) |>
  arrange(eid, title) |>
  select(-id) |>
  pivot_wider(names_from = title, values_from = has)
```

```{r read_all_cn}
ukb_47s <- read_csv(
  file.path(root_fd, 'data', 'cn', 'ukb_47s.csv'),
  col_types = cols())
```

```{r read_demographics}
ukb_demo <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_demo.csv'),
  col_types = cols())
```

```{r identify_cn_other_eth}
ukb_47s_o <- ukb_demo %>%
  inner_join(ukb_47s, by = 'eid') %>%
  filter(Eth_Back != 1001)
```

```{r read_genotyping_arrays}
ukb_geno_array <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_array.csv'),
  col_types = cols()
)
```

```{r read_pcs}
ukb_pcs <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_pc.csv'),
  col_types = cols()
)
```

```{r read_seq_rel}
ukb_sr <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_sr.csv'),
  col_types = cols()
)
```

```{r read_adj_tel_length}
ukb_atl <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_adj_tl.csv'),
  col_types = cols()
)
```

```{r read_assessment_centre}
ukb_ac <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_ac.csv'),
  col_types = cols()
)
```

```{r identify_kidney_diseases_other_eth}
ukb_group_kidney_o <- ukb_kidney_d_all_eth |>
  filter(eid %in% ukb_47s_o$eid) |>
  pivot_longer(-eid) |>
  group_by(name) |>
  separate(name, into = c('code'), sep = ' ', extra = 'drop') |>
  filter(code %in% c(ukb_glom_d_c, ukb_fail_kidney_c, ukb_urolithiasis_c)) |>
  mutate(kidney_g = case_when(
    code %in% ukb_glom_d_c ~ 'Glomerular D.',
    code %in% ukb_fail_kidney_c ~ 'Kidney Fail.', 
    code %in% ukb_urolithiasis_c ~ 'Urolithiasis')) |>
  group_by(eid, kidney_g) |>
  summarise(`Kidney D.` = any(value), .groups = 'drop') |>
  pivot_wider(names_from = 'kidney_g', values_from = `Kidney D.`) |>
  ungroup()
```

```{r construct_kidney_disease_covars_other_eth}
ukb_covars_kidney_o <- ukb_47s_o %>%
    filter(eid %in% c(ukb_geno_array %>% 
                        filter(Array == 1) %>% pull(eid))) %>%
    inner_join(ukb_pcs, by = 'eid') %>%
    inner_join(ukb_sr, by = 'eid') %>%
    inner_join(ukb_atl, by = 'eid') %>%
    inner_join(ukb_ac %>% select(-coding), by = 'eid') %>%
    inner_join(ukb_group_kidney_o, by = 'eid') %>%
    mutate(`Age Sq` = Age * Age) %>%
    select(eid, Sex, 
           starts_with('Age'), 
           Eth_Back,
           contains('Centre'),
           paste0('PC', 1:10),
           starts_with('Adj'),
           all_of(kidney_gs),
           CN)

ukb_covars_kidney_o[is.na(ukb_covars_kidney_o) | 
                      ukb_covars_kidney_o == "Inf"] <- NA
```

```{r scale_kidney_disease_covars_other_eth}
ukb_group_kidney_o_scaled <- ukb_covars_kidney_o |>
  select(-eid) |>
  mutate(across(where(is.numeric), scale)) 
```

```{r fit_kidney_diseases_other_eth, warning = F}
ukb_group_kidney_o_sig <- suppressWarnings(
  bind_rows(
  lapply(
    X = kidney_gs,
    FUN = function(x) {
      kidney_gs_other <- setdiff(kidney_gs, x)
      broom::tidy(
          glm(as.formula(paste0('`', stringr::str_trim(x), '` ~ . ')),
              ukb_group_kidney_o_scaled |> select(-all_of(kidney_gs_other)), 
          family = 'binomial'),
          conf.int = TRUE) |>
        mutate(lpval = -log10(p.value),
               scoeff = sign(estimate))  |>
          filter(term == 'CN') |>
          select(-term) |>
          mutate(term = x) |>
          relocate(term)
    })))
```

```{r display_odds_ratios_kidney_diseases_other_eth}
ukb_group_kidney_o_sig |>
  mutate(OR = exp(estimate), OR_l = exp(conf.low), OR_h = exp(conf.high))
```

## Comparison of Proteomics Profiles

### Read proteomics values

```{r read_proteomics_wgs}
proteomics_brit <- read_csv(
  file.path(root_fd,
            'proteomics', 'data', 'proteins', 'proteomics_british.csv'),
  col_types = cols())
```

```{r extract_protein_names}
proteins <- colnames(proteomics_brit)[-1]
```

```{r read_plate_ids}
prot_plate_ids <- read_csv(
  file.path(root_fd, 'data', 'releases', 'pheno_10.csv'),
  col_types = cols()) |>
  select(eid, starts_with('30901')) |>
  mutate(`Plate ID` = coalesce(
    `30901-0.0`, `30901-1.0`, `30901-2.0`, `30901-3.0`)) |>
  select(eid, `Plate ID`) |>
  filter(!is.na(`Plate ID`)) |>
  group_by(`Plate ID`)
```

```{r read_assessment_centre}
ukb_ac <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_ac.csv'),
  col_types = cols())
```

```{r read_pcs}
ukb_pcs <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_pc.csv'),
  col_types = cols())
```

```{r read blood_pcs}
ukb_blood_pc <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_blood_pc.csv'),
  col_types = cols())
```

```{r read_genotyping_array}
ukb_geno_array <- 
  read_csv(file.path(root_fd, 'data', 'pheno', 'ukb_array.csv'),
            col_types = cols())
```

```{r subset_genotyping_array}
ukb_geno_array_wb <- ukb_geno_array |>
  inner_join(ukb_demo, by = 'eid') |>  
  filter(Eth_Back_Gen == 1, Eth_Back == 1001) |>
  select(eid, Array, Batch)
```

```{r construct_covars_wb}
ukb_covars_wb <- ukb_demo |>
  inner_join(ukb_geno_array_wb, by = 'eid') |>
  filter(Array == 1) |>
  inner_join(ukb_pcs, by = 'eid') |>
  inner_join(ukb_blood_pc, by = 'eid') |>
  inner_join(ukb_ac |> select(-coding), by = 'eid') |>
  mutate(`Age Sq` = Age * Age) |>
  select(eid, Sex,
         starts_with('Age'), 
         contains('Centre'),
         paste0('PC', 1:10),
         starts_with('Blood')) 
```

```{r construct_proteomics_covars}
ukb_proteomics_covar <- ukb_egfr_cr_cc |>
  rename(eGFR = starts_with('eGFR')) |>
  inner_join(ukb_br, by = 'eid') |>
  inner_join(ukb_K_lipo_A |> select(eid, starts_with('Lipo')),
             by = 'eid') |>
  inner_join(ukb_covars_wb, by = 'eid') |>
  filter(eid %in% proteomics_brit$eid) |>
  inner_join(prot_plate_ids, by = 'eid')
```

```{r save_proteomics_covar}
ukb_proteomics_covar |>
  write_csv(file.path(
    root_fd, 'proteomics', 'data', 'proteins', 'protein_covars.csv'),
    na = '')
```

### Fit Model for 18S Ratio

### rDNA CN proteomics associations

```{r fit_protein_cn_function}
fit_protein_cn_plate <- function(df, protein = 'aarsd1') {
  aux <- df |>
    inner_join(proteomics_brit |> select(eid, all_of(protein)),
               by = 'eid')
  broom::tidy(
    lm(
      as.formula(paste0(protein,  '~ .')),
      aux |>
        select(Sex, starts_with('Age'), contains('Centre'), 
             starts_with('PC'), starts_with('Adj'),
             all_of(protein), `Plate ID`, CN) |>
             mutate(across(where(is.numeric), scale))),
    conf.int = TRUE) |> 
    mutate(lpval = -log10(p.value),
           scoeff = sign(estimate)) |>
    filter(term == 'CN') |>
    mutate(term = protein)
}
```

```{r fit_protein_cn}
proteins_cn_sig_plate <- bind_rows(
  lapply(
    X = proteins,
    FUN = function(x) {
      fit_protein_cn_plate(ukb_gwas_covars |>
                    inner_join(prot_plate_ids, by = 'eid'),
                  protein = x)
    }))
```

```{r save_proteomics_cn_associations}
proteins_cn_sig_plate |> write_csv(
  file.path(
    root_fd,  'proteomics', 'data', 'associations',
    'rdna_cn_plate_id_no_categories.csv'),
  na = '')
```

### Fit Model for Kidney Failure

```{r extract_kidney_disease, warning = F}
ukb_kidney_d_all <- read_csv(
 file.path(root_fd, 'data/pheno/ukb_kidney_d_all.csv'),
  col_types = cols()) |> 
  group_by(eid) |>
  pivot_longer(-eid, names_to = 'id', values_to = 'val') |>
    mutate(val = if_else(
      as.numeric(lubridate::year(val)) < 1910 |
        as.numeric(lubridate::year(val)) > 2036, NA, val)) |>
  mutate(has = ! is.na(val)) |>
  select(-val) |>
  group_by(id) |>
  separate(id, into = 'id', sep = '-', extra = 'drop') |>
  mutate(id = as.numeric(id)) |>
  inner_join(ukb_kidney_d_ids |> select(id, title), by = 'id') |>
  mutate(title = gsub('Date |first reported ', '', title)) |>
  arrange(eid, title) |>
  select(-id) |>
  pivot_wider(names_from = title, values_from = has)
```

```{r group_kf}
ukb_kf_all <- ukb_kidney_d_all |>
  pivot_longer(-eid) |>
  group_by(name) |>
  separate(name, into = c('code'), sep = ' ', extra = 'drop') |>
  filter(code %in% c(ukb_fail_kidney_c)) |>
  mutate(kidney_g = case_when(
    code %in% ukb_fail_kidney_c ~ 'Kidney Fail.',)) |>
  group_by(eid, kidney_g) |>
  summarise(`Kidney D.` = any(value), .groups = 'drop') |>
  pivot_wider(names_from = 'kidney_g', values_from = `Kidney D.`) |>
  ungroup()
```

```{r save_kf}
ukb_kf_all |> 
  write_csv(file.path(root_fd, 'data', 'pheno', 'ukb_kf_all.csv'),
            na = '')
```

```{r read_proteomics_wgs}
proteomics_brit <- read_csv(
  file.path(root_fd,
            'proteomics', 'data', 'proteins', 'proteomics_british.csv'),
  col_types = cols())
```

```{r construct_kidney_fail_proteomics_covars}
ukb_group_kf_all_covar <- ukb_kf_all |>
  inner_join(ukb_covars_wb, by = 'eid') |>
  filter(eid %in% proteomics_brit$eid)
```

```{r fit_protein_kidney_fail_function}
fit_protein_kf_plate <- function(df, protein = 'aarsd1', blood = T) {
  
  aux <- df
  if (! blood ){
    aux <- aux |>
      select(-starts_with('Blood'))
  }
    
  aux <- aux |>
    inner_join(proteomics_brit |> select(eid, all_of(protein)),
               by = 'eid')
  broom::tidy(
    lm(
      as.formula(paste0(protein,  '~ .')),
      aux |>
        select(Sex, starts_with('Age'), starts_with('PC'), 
             all_of(protein) ,`Kidney Fail.`, 
             any_of(starts_with('Blood')),
             starts_with('Plate')) |>
             mutate(across(where(is.numeric), scale))),
    conf.int = TRUE) |> 
    mutate(lpval = -log10(p.value),
           scoeff = sign(estimate)) |>
    filter(grepl('Kidney', term)) |>
    mutate(term = protein)
}
```

```{r fit_protein_kidney_fail}
proteins_kf_all_sig_plate <- bind_rows(
    lapply(X = proteins,
           FUN = function(x) 
             fit_protein_kf_plate(
               df = ukb_group_kf_all_covar |>
                 inner_join(prot_plate_ids, by = 'eid'),
               protein = x, blood = F)))
```

```{r save_proteomics_kf_associations}
proteins_kf_all_sig_plate |> write_csv(
  file.path(
    root_fd,  'proteomics', 'data', 'associations',
    'kidney_fail_all_wb_plate_id_no_categories.csv'),
  na = '')
```


### Fit Model for Other Traits

This might need to be run in HPC servers and the results read in afterwards.

```{r fit_protein_function}
fit_protein_plate <- function(
    df, the_var = 'NLR', protein = 'aarsd1', blood = T) {
  
  aux <- df
  if (! blood ){
    aux <- aux |>
      select(-starts_with('Blood'))
  }
    
  aux <- aux |>
    inner_join(proteomics_brit |> select(eid, all_of(protein)),
               by = 'eid')
  broom::tidy(
    lm(
      as.formula(paste0(protein,  '~ .')),
      aux |>
        select(Sex, starts_with('Age'), starts_with('PC'), 
             all_of(protein), all_of(the_var), 
             any_of(starts_with('Blood')),
             starts_with('Plate')) |>
             mutate(across(where(is.numeric), scale))),
    conf.int = TRUE) |> 
    mutate(lpval = -log10(p.value),
           scoeff = sign(estimate)) |>
    filter(grepl(the_var, term)) |>
    mutate(protein = protein, term = gsub('`', '', term)) |>
    relocate(protein, .after = 1L)
}
```

```{r define_phenotypes_to_fit_proteomics}
protein_vars <- c('NLR', 'PLR', 'LMR', 'SII', 'Lipoprotein A')
```

```{r fit_proteomics, eval = F}
proteins_sig_plate <- bind_rows(
  bind_rows(
    lapply(X = proteins,
           FUN = function(x) 
             lapply(
               X = protein_vars,
               FUN =  function(y)
                 fit_protein_plate(
                   df = ukb_proteomics_covar, 
                   protein = x, the_var = y, blood = T)))) |>
    mutate(`Blood PCs` = 'Yes'),
  bind_rows(
    lapply(X = proteins,
           FUN = function(x) 
             lapply(
               X = protein_vars,
               FUN = function(y)
                 fit_protein_plate(
                   df = ukb_proteomics_covar,
                   protein = x, the_var = y, blood = F)))) |>
    mutate(`Blood PCs` = 'No')
)
```

Reading results obtained on servers

```{r read_proteomic_results}
prot_control_fd <- file.path(
  root_fd, 'proteomics', 'data', 'associations_control')
prot_control_fns <- list.files(prot_control_fd)
prot_control <- bind_rows(
  lapply(
    X = prot_control_fns,
    FUN = function(x) read_csv(
      file.path(prot_control_fd, x),
      col_types = cols()) ) )
```

```{r extract_proteomic_results}
prot_control_betas <- prot_control |>
  filter(`Blood PCs` == 'No') |>
  select(term, protein, estimate) |>
  group_by(protein) |>
  pivot_wider(names_from = term, values_from = estimate) |>
  ungroup() |>
  select(-protein)
```

### Combine

```{r combine_proteomics_results}
prot_control_cn <- prot_control |>
  filter(`Blood PCs` == 'No') |>
  select(-`Blood PCs`) |>
  group_by(term) |>
  mutate(fdr = p.adjust(p.value, method = 'fdr')) |>
  filter(fdr < 0.01) |>
  inner_join(rdna_cn_prot_sig |> 
               select(protein = term, 
                      estimate_cn = estimate),
             relationship = "many-to-many",
             by = 'protein')
```

```{r define_proteomics_plot_data}
prot_control_cn_plot <- bind_rows(
  prot_control_cn, 
  proteins_kf_all_sig_plate |>
  mutate(protein = term, term = 'Kidney Failure') |>
  group_by(term) |>
  mutate(fdr = p.adjust(p.value, method = 'fdr')) |>
  filter(fdr < 0.01) |>
  inner_join(rdna_cn_prot_sig |> 
               select(protein = term, 
                      estimate_cn = estimate),
             by = 'protein')) |>
  select(term, protein, estimate, estimate_cn) |>
  filter(! term %in% c('pm10')) |>
  mutate(name = paste0(term, ' [N = ', 
                       format(n_distinct(protein),
                              big.mark = ","), ']'))
```

```{r plot_proteomics_first_row}
prot_control_cn_plot_row1 <- prot_control_cn_plot |>
  filter(term %in% c('NLR', 'PLR', 'SII')) |>
  mutate(term = factor(term, levels = c('NLR', 'PLR', 'SII'))) |>
  ggplot(aes(x = estimate, y = estimate_cn)) +
  geom_point(shape = 21, fill = 'grey60', alpha = 0.3) +
  geom_smooth(method = 'lm', colour = 'black', alpha = 0.1, linewidth = 0.75) +
  ggpubr::stat_cor(label.y.npc = 0.1,
                   aes(label = cor_format(after_stat(r), after_stat(p)))) +
  facet_grid(cols = vars(name)) +
  scale_x_continuous(limits = c(-0.55, 0.55),
                     breaks = c(-0.5, 0, 0.5)) +
  scale_y_continuous(limits = c(-0.04, 0.04),
                     breaks = c(-0.04, 0, 0.04)) +
  theme_pub() +
  labs(x = 'Standardised Beta for Variable of Interest',
       y = 'Standardised Beta for 18S Ratio') +
  theme(
    aspect.ratio = 1)
```

```{r plot_proteomics_second_row}
prot_control_cn_plot_row2 <- prot_control_cn_plot |>
  filter(! term %in% c('NLR', 'PLR', 'SII', 'LMR')) |>
  ggplot(aes(x = estimate, y = estimate_cn)) +
  geom_point(shape = 21, fill = 'grey60', alpha = 0.3) +
  geom_smooth(method = 'lm', colour = 'black', alpha = 0.1, linewidth = 0.75) +
  ggpubr::stat_cor(label.y.npc = 0.1,
                   aes(label = cor_format(after_stat(r), after_stat(p)))) +
  facet_grid(cols = vars(name)) +
  scale_x_continuous(limits = c(-1.1, 1.1),
                     breaks = c(-1, 0, 1)) +
  scale_y_continuous(limits = c(-0.04, 0.04),
                     breaks = c(-0.04, 0, 0.04)) +
  theme_pub() +
  labs(x = 'Standardised Beta for Variable of Interest',
       y = 'Standardised Beta for 18S Ratio') +
  theme(
    aspect.ratio = 1)
```

```{r save_proteomics_plot_second_row}
prot_control_cn_plot_row2 |>
  ggsave(
    filename = file.path('..', 'figures', 
                         '4.E.- Proteomic profiles, 2nd row.png'),
    width = 8, height = 6, dpi = 300)
```

```{r save_proteomics_plot_both_rows}
prot_control_cn_plot_rows <- (
  prot_control_cn_plot_row1  +
    theme(axis.title = element_blank())) /
  prot_control_cn_plot_row2 &
  ylab(NULL) & theme(panel.spacing = unit(0.4, 'lines'))
prot_control_cn_plot_rows |>
  ggsave(
    filename = file.path('..', 'figures', '4.E.- Proteomic profiles.png'),
    width = 8.5, height = 6.5, dpi = 300)
```

## Supplementary: eGFR Comparison according to Kidney Failure status

```{r join_egfr_kf}
ukb_egfr_kf <- ukb_egfr_cr_cc |>
  inner_join(ukb_group_kidney |> select(eid, starts_with('Kid')),
             by = 'eid') |>
  mutate(`Kidney Failure` = if_else(`Kidney Fail.`, 'Yes', 'No'))
```

```{r plot_egfr_kf}
ukb_egfr_kf_p <- ukb_egfr_kf |>
  ggplot(aes(y = `eGFRcr+cc`, x = `Kidney Failure`,
             fill = `Kidney Failure`)) +
  geom_boxplot(
    notch = TRUE,
    outlier.alpha = 0.3,
    outlier.shape = 21,
    linewidth = 0.75, alpha = 0.8) +
  labs(y = 'eGFR<sub>cr+cc</sub>') +
  scale_fill_manual('', values = c('Yes' = 'tan4',
                                   'No' = 'peachpuff')) +
  ggpubr::stat_compare_means(
    comparisons = list(c('Yes', 'No'))) +
  lims(y = c(0, 160)) +
  theme_pub() +
  theme(axis.title.y = element_markdown(),
        aspect.ratio = 1,
        legend.position = 'none')
```

```{r save_egfr_kf_p}
ukb_egfr_kf_p |>
  ggsave(
    filename = '../figures_supp/eGFR v KF.png',
    width = 6, height = 6, dpi = 300)
```

## Proteomics Betas Signs

```{r extract_proteomics_betas}
prot_betas <- prot_control_cn_plot |>
  filter(! term %in% c('Lipoprotein A', 'LMR')) |>
  mutate(type = if_else(term %in% c('NLR', 'PLR', 'SII'),
                        'Blood\nComposition',
                        'Renal\nFunction'))
```

```{r plot_proteomics_betas_panel_1}
prot_betas_p1 <- prot_betas |> 
  mutate(term = gsub(' ', '\n', term)) |>
  ggplot(aes(x = estimate, y = term, fill = term)) + 
  geom_vline(xintercept = 0, alpha = 0.4, colour = 'grey50') +
  geom_boxplot(notch = TRUE, 
               outlier.alpha = 0.3, outlier.shape = 21,
               alpha = 0.8, linewidth = 0.6) +
  facet_grid(rows = vars(type), scales = 'free') +
  scale_fill_manual(values = c(
    'eGFR' = 'tan2', 'Kidney\nFailure' = 'tan4',
    'SII' = 'firebrick4', 'PLR' = 'firebrick2', 'NLR' = 'lightcoral')) +
  theme_pub() +
  xlim(c(-1, 1)) +
  labs(x = 'FDR-significant Std. Betas' , 
       y = 'Variable of Interest') +
  theme(legend.position = 'none', strip.text = element_blank())
```

```{r save_proteomics_betas_plot_panel_1}
prot_betas_p1 |>
  ggsave(
    filename = file.path('..', 'figures', '4.F.- Proteomics betas, 1.png'),
    width = 5, height = 5, dpi = 300)
```

```{r plot_proteomics_betas_panel_2}
prot_betas_p2 <- prot_betas |> 
  mutate(term = gsub(' ', '\n', term),
         type = forcats::fct_rev(type)) |>
  ggplot(aes(x = estimate, y = term, fill = term)) + 
  geom_vline(xintercept = 0, alpha = 0.4, colour = 'grey50') +
  geom_boxplot(notch = TRUE, 
               outlier.alpha = 0.3, outlier.shape = 21,
               alpha = 0.8, linewidth = 0.6) +
  facet_grid(rows = vars(type), scales = 'free') +
  scale_fill_manual(values = c(
    'eGFR' = 'tan2', 'Kidney\nFailure' = 'tan4',
    'SII' = 'firebrick4', 'PLR' = 'firebrick2', 'NLR' = 'lightcoral')) +
  theme_pub() +
  xlim(c(-0.5, 0.5)) +
  labs(x = 'FDR-significant Std. Betas' , 
       y = 'Variable of Interest') +
  theme(legend.position = 'none', strip.text = element_blank())
```

```{r save_proteomics_betas_plot_panel_2}
prot_betas_p2 |>
  ggsave(
    filename = file.path('..', 'figures', '4.F.- Proteomics betas, 2.png'),
    width = 5, height = 5, dpi = 300)
```
