---
title: "Supplementary Figures"
---

## Setup

```{r setup}
source('0.0.- Setup.R')
```

## S12.- Correlation between blood counts

```{r read_blood_counts}
ukb_blood_counts <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_blood_counts.csv'),
  col_types = cols())
```

```{r correlate_blood_counts}
ukb_bc_cor_pr <- cor(ukb_blood_counts %>%
      na.omit() %>%
      filter(!is.na(`Erythrocyte #`), ! is.na(`Platelet #`)) %>%
      select(ends_with('#'))) %>%
  as_tibble(rownames = 'cell1') |>
  pivot_longer(-cell1, names_to = 'cell2', values_to = 'pr') %>%
  filter(cell1 < cell2)
```

```{r plot_blood_count_correlations}
ukb_bc_cor_p <- ukb_bc_cor_pr %>%
  ggplot(aes(x = cell1, y = cell2, fill = pr)) +
  geom_tile(colour = 'black', size = 0.5) +
  scale_x_discrete(position = 'top') +
  scale_fill_gradientn(
    "Pearson's R",
    limits = c(-1, 1),
    colours=c("red4", "indianred", "white", "steelblue", "blue4")) +
  labs(x = '', y = '') +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.title = element_text(size = 14,
                                    face = 'bold'),
        legend.text = element_text(size = 12), 
        panel.border = element_blank(),
        aspect.ratio = 1,
        panel.grid = element_blank(),
        axis.ticks = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 0)) +
  guides(fill = guide_legend(title.position="top"))
```

```{r save_blood_count_correlations_plot}
ggsave(
  plot = ukb_bc_cor_p,
  filename = file.path('..', 'figures_supp', 
                       'S12.- Correlation between blood counts.png'), 
  height = 7, width = 7.5)
```

```{r correlate_blood_counts_with_sig}
corrFunc <- function(var1, var2, data) {
  result <- broom::tidy(cor.test(data[[var1]], data[[var2]],
                                 method = 'pearson'))
  bind_cols(
    tibble(var1, var2),
    result |> select(estimate, pval = p.value, statistic))
}

cor_df <- ukb_blood_counts |>
  na.omit() |>
  filter(!is.na(`Erythrocyte #`), ! is.na(`Platelet #`)) |>
  select(ends_with('#'))

## Pairs of variables for which we want correlations
vars <- expand.grid(
  v1 = colnames(cor_df),
  v2 = colnames(cor_df),
  stringsAsFactors = F) |>
  filter(v1 > v2) |>
  arrange(v1, v2)

# Apply corrFunc to all rows of vars
(corrs <- do.call(rbind,
                 mapply(corrFunc, 
                        vars$v1, 
                        vars$v2, 
                        MoreArgs=list(data=cor_df), 
                        SIMPLIFY=FALSE)))
```

## S20.- Location of 7980 analogues in BWA-MEM alignments

```{r read_locations}
gbr_loc_als <- read_csv(
  file.path(root_fd, '..', '1000g', 'data', 'gbr_loc_als.csv'), 
  col_types = cols()) 
```

```{r calculate_location_frequencies}
gbr_loc_freqs <- gbr_loc_als %>%
    complete(
        sample,
        nesting(contig, strand),
        fill = list(num_reads = 0)) |>
  group_by(sample, strand) |>
  mutate(freq_reads = num_reads / sum(num_reads))
```

```{r plot_locations}
gbr_loc_freqs_p <- gbr_loc_freqs |>
  ungroup() |>
  mutate(strand = if_else(strand == 'f', 'Forward', 'Reverse'),
         contig = forcats::fct_rev(contig)) |>
  ggplot(aes(y = contig,
             x = freq_reads,
             fill = strand)) +
  geom_boxplot(outlier.shape = 21,
               outlier.alpha = 0.5,
               position = position_dodge(width = 0.9)) +
  scale_fill_brewer('Alignment strand in bowtie2', palette = 'Reds') +
  labs(y = 'Hg38 contig', x = 'Frequency in BWA-MEM alignments') +
  theme_pub() +
  theme(
    legend.position = 'bottom',
    legend.title = element_text(size = 14, face = 'bold'),
    legend.text = element_text(size = 12)
  )
```

```{r save_location_plots}
ggsave(
  plot = gbr_loc_freqs_p,
  filename = file.path('..', 'figures_supp',
                       'S20.- Location of 7980 reads in BWA-MEM.png'), 
  height = 12, width = 8)
```

