---
title: "Supplementary Figures"
---

## Setup

```{r setup}
source('0.0.- Setup.R')
```

```{r read_cn_wb}
ukb_47s_wb <- read_csv(
  file.path(root_fd, 'data', 'cn', 'ukb_47s_wb.csv'),
  col_types = cols())
```

```{r read_adj_cn_wb}
ukb_adj_47s_diff <- read_csv(
    file.path(root_fd, 'data', 'cn', 'ukb_adj_cn_wb.csv'),
    col_types = cols())
```

```{r read_seq_rel}
ukb_sr <- read_csv(
  file.path(
    root_fd, 'data', 'pheno', 'ukb_sr.csv'),
  col_types = cols())
```

## S7.- Twins Sharing Sequencing Release

```{r find_twins_cn_sharing_seq_rel}
ukb_twins_cn_sr <- read_csv(
    file.path(root_fd, 'data', 'pheno', 'ukb_rel_first_type.csv'),
    col_types = cols()) |>
  filter(rel_type == 'Twin') |>
  inner_join(ukb_sr, by = c('eid_o'= 'eid')) |>
  inner_join(ukb_sr, by = c('eid_y' = 'eid'), suffix = c('_o', '_y')) |>
  filter(`Seq Release_o` == `Seq Release_y`) |>
  inner_join(ukb_47s_wb |> select(eid, CN), by = c('eid_o' = 'eid')) |>
  inner_join(ukb_47s_wb |> select(eid, CN),
             by = c('eid_y' = 'eid'), suffix = c('_o', '_y')) |>
  rename(sr = `Seq Release_o`)
```

```{r plot_twins_cn_sharing_seq_rel}
ukb_twins_cn_sr_p <- ukb_twins_cn_sr |>
  ggplot(aes(x = CN_o, y = CN_y)) +
  geom_point(aes(shape = sr, fill = sr),
             alpha = 0.5, size = 2,
             stroke = 0.75) +
  geom_smooth(method = 'lm', size = 0.5, colour = 'grey20', alpha = 0.1) +
  facet_grid(cols = vars(sr)) +
  ggpubr::stat_cor(aes(
    label = cor_format(..r.., ..p..)),
    size = 5, label.y.npc = 0.96) +
  scale_x_continuous(limits = c(0, 4.75e-4),
                     breaks = c(0, 2e-4, 4e-4),
                     labels = fancy_scientific) +
  scale_y_continuous(limits = c(0, 4.75e-4),
                     breaks = c(0, 2e-4, 4e-4),
                     labels = fancy_scientific) +
  labs(x = '18S Ratio for Twin 1',
      y = '18S Ratio for Twin 2') +
  scale_shape_manual(
    'Sequencing Batch',
    values = c(21, 22, 23)) +
  scale_fill_brewer(
    'Sequencing Batch',
    palette = 'Greys') +
  theme_pub() +
  theme(aspect.ratio = 1,
        legend.position = 'none')
```

```{r save_twins_cn_sharing_seq_rel_plot}
ggsave(
  plot = ukb_twins_cn_sr_p,
  filename = file.path('..', 'figures_supp', 'S7.- CN Twins, batch.png'), 
  height = 5, width = 10)
```

```{r plot_twins_cn_sharing_seq_rel_joint}
ukb_twins_cn_sr_join_p <- ukb_twins_cn_sr |>
  ggplot(aes(x = CN_o, y = CN_y)) +
  geom_point(aes(shape = sr, fill = sr),
             alpha = 0.5, size = 2,
             stroke = 0.75) +
  geom_smooth(method = 'lm', size = 0.5, 
              colour = 'grey20', alpha = 0.1) +
  ggpubr::stat_cor(aes(
    label = cor_format(..r.., ..p..)),
    size = 5, label.y.npc = 0.96) +
  scale_x_continuous(limits = c(0, 4.75e-4),
                     breaks = c(0, 2e-4, 4e-4),
                     labels = fancy_scientific) +
  scale_y_continuous(limits = c(0, 4.75e-4),
                     breaks = c(0, 2e-4, 4e-4),
                     labels = fancy_scientific) +
  scale_shape_manual(
    'Sequencing Batch',
    values = c(21, 22, 23)) +
  scale_fill_brewer(
    'Sequencing Batch',
    palette = 'Greys') +
  labs(
    x = '18S Ratio for Twin 1',
    y = '18S Ratio for Twin 2') +
  theme_pub() +
  theme(aspect.ratio = 1,
        legend.title = element_text(size = 14, face = 'bold'),
        legend.text = element_text(size = 12))
```

```{r save_twins_cn_sharing_seq_rel_joint_plot}
ggsave(
  plot = ukb_twins_cn_sr_join_p,
  filename = file.path('..', 'figures_supp', 'S7.- CN Twins.png'), 
  height = 5, width = 7)
```

## S10.- Correlation between blood counts

```{r read_blood_counts}
ukb_blood_counts <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_blood_counts.csv'),
  col_types = cols())
```

```{r correlate_blood_counts}
ukb_bc_cor_pr <- cor(ukb_blood_counts %>%
      na.omit() %>%
      filter(!is.na(`Erythrocyte #`), ! is.na(`Platelet #`)) %>%
      select(ends_with('#'))) %>%
  as_tibble(rownames = 'cell1') |>
  pivot_longer(-cell1, names_to = 'cell2', values_to = 'pr') %>%
  filter(cell1 < cell2)
```

```{r plot_blood_count_correlations}
ukb_bc_cor_p <- ukb_bc_cor_pr %>%
  ggplot(aes(x = cell1, y = cell2, fill = pr)) +
  geom_tile(colour = 'black', size = 0.5) +
  scale_x_discrete(position = 'top') +
  scale_fill_gradientn(
    "Pearson's R",
    limits = c(-1, 1),
    colours=c("red4", "indianred", "white", "steelblue", "blue4")) +
  labs(x = '', y = '') +
  theme_pub() +
  theme(legend.position = 'bottom',
        legend.title = element_text(size = 14,
                                    face = 'bold'),
        legend.text = element_text(size = 12), 
        panel.border = element_blank(),
        aspect.ratio = 1,
        panel.grid = element_blank(),
        axis.ticks = element_blank(),
        axis.text.x = element_text(angle = 45, hjust = 0)) +
  guides(fill = guide_legend(title.position="top"))
```

```{r save_blood_count_correlations_plot}
ggsave(
  plot = ukb_bc_cor_p,
  filename = file.path('..', 'figures_supp', 
                       'S12.- Correlation between blood counts.png'), 
  height = 7, width = 7.5)
```


```{r correlate_blood_counts_with_sig}
corrFunc <- function(var1, var2, data) {
  result <- broom::tidy(cor.test(data[[var1]], data[[var2]],
                                 method = 'pearson'))
  bind_cols(
    tibble(var1, var2),
    result |> select(estimate, pval = p.value, statistic))
}

cor_df <- ukb_blood_counts |>
  na.omit() |>
  filter(!is.na(`Erythrocyte #`), ! is.na(`Platelet #`)) |>
  select(ends_with('#'))

## Pairs of variables for which we want correlations
vars <- expand.grid(
  v1 = colnames(cor_df),
  v2 = colnames(cor_df),
  stringsAsFactors = F) |>
  filter(v1 > v2) |>
  arrange(v1, v2)

# Apply corrFunc to all rows of vars
(corrs <- do.call(rbind,
                 mapply(corrFunc, 
                        vars$v1, 
                        vars$v2, 
                        MoreArgs=list(data=cor_df), 
                        SIMPLIFY=FALSE)))
```

## S16.- Proteomics

### Association with 18S Ratio

```{r read_proteomics_wgs}
proteomics <- read_csv(
  file.path(root_fd, 'proteomics', 'data', 'proteins',
            'proteomics_british_wgs.csv'),
  col_types = cols())
```

```{r read_covariates}
ukb_gwas_covars <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_gwas_covars.csv'),
  col_types = cols())
```

```{r extract_protein_names}
proteins <- colnames(proteomics)[-1]
```

```{r read_olink_categories}
protein_cats <- readxl::read_excel(
  file.path(root_fd,
            'proteomics', 'data', 'proteins',
            'olink-explore-3072-assay-list-2023-06-08.xlsx'),
  skip = 2) |>
  mutate(protein = tolower(`Gene name`)) |>
  filter(protein %in% proteins) |> distinct() |>
  rename(Category = `Explore 384 panel`)
```

```{r read_plate_ids}
prot_plate_ids <- read_csv(
  file.path(root_fd, 'data', 'releases', 'pheno_10.csv'),
  col_types = cols()) |>
  select(eid, starts_with('30901')) |>
  mutate(`Plate ID` = coalesce(
    `30901-0.0`, `30901-1.0`, `30901-2.0`, `30901-3.0`)) |>
  select(eid, `Plate ID`) |>
  filter(!is.na(`Plate ID`)) |>
  group_by(`Plate ID`)
```

```{r fit_protein_cn_function}
fit_protein_plate <- function(df, protein = 'aarsd1') {
  aux <- df |>
    inner_join(proteomics |> select(eid, all_of(protein)),
               by = 'eid')
  broom::tidy(
    lm(
      as.formula(paste0(protein,  '~ .')),
      aux |>
        select(Sex, starts_with('Age'), contains('Centre'), 
             starts_with('PC'), starts_with('Adj'),
             all_of(protein), `Plate ID`, CN) |>
             mutate(across(where(is.numeric), scale))),
    conf.int = TRUE) |> 
    mutate(lpval = -log10(p.value),
           scoeff = sign(estimate)) |>
    filter(term == 'CN') |>
    mutate(term = protein)
}
```

```{r fit_protein_cn}
proteins_cn_sig_plate <- bind_rows(
  lapply(
    X = proteins,
    FUN = function(x) {
      fit_protein_plate(ukb_gwas_covars |>
                    inner_join(prot_plate_ids, by = 'eid'),
                  protein = x)
    }))
```

```{r add_categories_to_cn_associations}
proteins_cn_sig_cat_plate <- proteins_cn_sig_plate |>
  inner_join(protein_cats, by = c('term' = 'protein')) |>
  arrange(p.value) |>
  mutate(fdr = p.adjust(p.value, method = 'fdr')) |>
  relocate(Category, `Protein name`)
```

```{r save_protein_cn_associations}
write_csv(proteins_cn_sig_cat_plate,
          file.path(
            root_fd,
            'proteomics', 'data', 'associations', 'rdna_cn_plate_id.csv'),
          na = '')
```

### Association with Renal Failure

#### Read Input Data

```{r read_proteomics_wgs}
proteomics_brit <- read_csv(
  file.path('../../../',
            'proteomics/data/proteins/proteomics_british.csv'),
  col_types = cols())
```

```{r extract_kidney_disease, warning = F}
ukb_kidney_d_all <- read_csv(
 file.path(root_fd, 'data/pheno/ukb_kidney_d_all.csv'),
  col_types = cols()) |> 
  group_by(eid) |>
  pivot_longer(-eid, names_to = 'id', values_to = 'val') |>
    mutate(val = if_else(
      as.numeric(lubridate::year(val)) < 1910 |
        as.numeric(lubridate::year(val)) > 2036, NA, val)) |>
  mutate(has = ! is.na(val)) |>
  select(-val) |>
  group_by(id) |>
  separate(id, into = 'id', sep = '-', extra = 'drop') |>
  mutate(id = as.numeric(id)) |>
  inner_join(ukb_kidney_d_ids |> select(id, title), by = 'id') |>
  mutate(title = gsub('Date |first reported ', '', title)) |>
  arrange(eid, title) |>
  select(-id) |>
  pivot_wider(names_from = title, values_from = has)
```

```{r group_kidney_disease}
ukb_group_kidney_all <- ukb_kidney_d_all |>
  pivot_longer(-eid) |>
  group_by(name) |>
  separate(name, into = c('code'), sep = ' ', extra = 'drop') |>
  filter(code %in% c(ukb_glom_d_c, ukb_fail_kidney_c, ukb_urolithiasis_c)) |>
  mutate(kidney_g = case_when(
    code %in% ukb_glom_d_c ~ 'Glomerular D.',
    code %in% ukb_fail_kidney_c ~ 'Kidney Fail.', 
    code %in% ukb_urolithiasis_c ~ 'Urolithiasis')) |>
  group_by(eid, kidney_g) |>
  summarise(`Kidney D.` = any(value), .groups = 'drop') |>
  pivot_wider(names_from = 'kidney_g', values_from = `Kidney D.`) |>
  ungroup()
```

#### Generate Covariates

```{r read_assessment_centre}
ukb_ac <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_ac.csv'),
  col_types = cols())
```

```{r read_demographics}
ukb_demo <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_demo.csv'),
  col_types = cols())
```

```{r read_pcs}
ukb_pcs <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_pc.csv'),
  col_types = cols())
```

```{r read blood_pcs}
ukb_blood_pc <- read_csv(
  file.path(root_fd, 'data', 'pheno', 'ukb_blood_pc.csv'),
  col_types = cols())
```

```{r read_genotyping_array}
ukb_geno_array <- 
  read_csv(file.path(root_fd, 'data', 'pheno', 'ukb_array.csv'),
            col_types = cols())
```

```{r subset_genotyping_array}
ukb_geno_array_wb <- ukb_geno_array |>
  inner_join(ukb_demo, by = 'eid') |>  
  filter(Eth_Back_Gen == 1, Eth_Back == 1001) |>
  select(eid, Array, Batch)
```

```{r construct_covars_wb}
ukb_covars_wb <- ukb_demo |>
  inner_join(ukb_geno_array_wb, by = 'eid') |>
  filter(Array == 1) |>
  inner_join(ukb_pcs, by = 'eid') |>
  inner_join(ukb_blood_pc, by = 'eid') |>
  inner_join(ukb_ac |> select(-coding), by = 'eid') |>
  mutate(`Age Sq` = Age * Age) |>
  select(eid, Sex,
         starts_with('Age'), 
         contains('Centre'),
         paste0('PC', 1:10),
         starts_with('Blood')) 
```

```{r construct_kidney_fail_proteomics_covars}
ukb_group_kidney_fail_all_covar <- ukb_group_kidney_all |>
  select(eid, `Kidney Fail.`) |>
  inner_join(ukb_covars_wb, by = 'eid') |>
  filter(eid %in% proteomics_brit$eid)
```

#### Fit Linear Model

```{r fit_protein_kidney_fail_function}
fit_protein_kidney_fail_plate <- function(df, protein = 'aarsd1', blood = T) {
  
  aux <- df
  if (! blood ){
    aux <- aux |>
      select(-starts_with('Blood'))
  }
    
  aux <- aux |>
    inner_join(proteomics |> select(eid, all_of(protein)),
               by = 'eid')
  broom::tidy(
    lm(
      as.formula(paste0(protein,  '~ .')),
      aux |>
        select(Sex, starts_with('Age'), starts_with('PC'), 
             all_of(protein) ,`Kidney Fail.`, 
             any_of(starts_with('Blood')),
             starts_with('Plate')) |>
             mutate(across(where(is.numeric), scale))),
    conf.int = TRUE) |> 
    mutate(lpval = -log10(p.value),
           scoeff = sign(estimate)) |>
    filter(grepl('Kidney', term)) |>
    mutate(term = protein)
}
```

```{r fit_protein_kidney_fail}
proteins_kidney_fail_all_sig_plate <- bind_rows(
  bind_rows(
    lapply(X = proteins,
           FUN = function(x) 
             fit_protein_kidney_fail_plate(
               df = ukb_group_kidney_fail_all_covar |>
                 inner_join(prot_plate_ids, by = 'eid'), 
               protein = x, blood = T))) |>
    mutate(`Blood PCs` = 'Yes'),
  bind_rows(
    lapply(X = proteins,
           FUN = function(x) 
             fit_protein_kidney_fail_plate(
               df = ukb_group_kidney_fail_all_covar |>
                 inner_join(prot_plate_ids, by = 'eid'),
               protein = x, blood = F))) |>
    mutate(`Blood PCs` = 'No')
)
```

```{r add_categories_to_kidney_fail_associations}
proteins_kidney_fail_all_sig_cat_plate <- proteins_kidney_fail_all_sig_plate |>
  inner_join(protein_cats, by = c('term' = 'protein')) |>
  arrange(p.value) |>
  mutate(fdr = p.adjust(p.value, method = 'fdr')) |>
  relocate(Category, `Protein name`)
```

```{r save_protein_kidney_fail_associations}
write_csv(proteins_kidney_fail_all_sig_cat_plate,
          file.path(
            root_fd,  'proteomics', 'data', 'associations',
            'kidney_fail_all_wb_plate_id.csv'),
          na = '')
```

### Compare Association Results

```{r compare_proteomics_kidney_fail_cn}
proteins_comp_all_plate <- proteins_kidney_fail_all_sig_cat_plate |>
  mutate(`Blood PCs` = paste0('Blood PCs: ', `Blood PCs`)) |>
  filter(fdr < 0.01) |>
  select(Category, term, estimate, `Blood PCs`) |>
  inner_join(proteins_cn_sig_cat_plate |>
               select(term, estimate),
             suffix = c('_kf', '_cn'),
             by = 'term')
```

```{r plot_proteomics_kidney_fail_cn_comparison}
proteins_comp_all_plate_no_blood_pcs_p <- proteins_comp_all_plate |>
  filter(grepl('No', `Blood PCs`)) |>
  ggplot(aes(x = estimate_kf, y = estimate_cn)) +
  geom_point(alpha = 0.3) +
  geom_smooth(method = 'lm', alpha = 0.1, colour = 'black',
              linewidth = 0.75) +
  ggpubr::stat_cor(label.y.npc = 0.1,
                   aes(label = cor_format(..r.., ..p..))) +
  labs(x = 'Standardised Beta for Kidney Failure',
       y = 'Standardised Beta for 18S Ratio') +
  scale_y_continuous(limits = c(-0.038, 0.038)) +
  scale_x_continuous(limits = c(-1.05, 1.05)) +
  theme_pub() +
  theme(aspect.ratio = 1)
```

```{r save_proteomics_kidney_fail_cn_comparison_plot}
ggsave(
  proteins_comp_all_plate_no_blood_pcs_p, 
  filename = file.path(
    '..', 'figures_supp', 
    'S16.- Proteomics KF vs rDNA CN, Plate ID, No Blood PCs.png'),
  width = 5, height = 5, dpi = 300)
```

## S17.- Location of 7980 analogues in BWA-MEM alignments

```{r read_locations}
gbr_loc_als <- read_csv(
  file.path(root_fd, '..', '1000g', 'data', 'gbr_loc_als.csv'), 
  col_types = cols()) 
```

```{r calculate_location_frequencies}
gbr_loc_freqs <- gbr_loc_als %>%
    complete(
        sample,
        nesting(contig, strand),
        fill = list(num_reads = 0)) |>
  group_by(sample, strand) |>
  mutate(freq_reads = num_reads / sum(num_reads))
```

```{r plot_locations}
gbr_loc_freqs_p <- gbr_loc_freqs |>
  ungroup() |>
  mutate(strand = if_else(strand == 'f', 'Forward', 'Reverse'),
         contig = forcats::fct_rev(contig)) |>
  ggplot(aes(y = contig,
             x = freq_reads,
             fill = strand)) +
  geom_boxplot(outlier.shape = 21,
               outlier.alpha = 0.5,
               position = position_dodge(width = 0.9)) +
  scale_fill_brewer('Alignment strand in bowtie2', palette = 'Reds') +
  labs(y = 'Hg38 contig', x = 'Frequency in BWA-MEM alignments') +
  theme_pub() +
  theme(
    legend.position = 'bottom',
    legend.title = element_text(size = 14, face = 'bold'),
    legend.text = element_text(size = 12)
  )
```


```{r save_location_plots}
ggsave(
  plot = gbr_loc_freqs_p,
  filename = file.path('..', 'figures_supp',
                       'S18.- Location of 7980 reads in BWA-MEM.png'), 
  height = 12, width = 8)
```

